---
title: '**Clearance of *Clostridioides difficile* colonization is associated with antibiotic-specific
  bacterial changes**'
csl: mbio.csl
fontsize: 11pt
output:
  pdf_document:
    includes:
      in_header: header.tex
    keep_tex: yes
  word_document: default
bibliography: references.bib
---


```{r knitr_settings, eval=TRUE, echo=FALSE, cache=FALSE, warning=FALSE, message=FALSE}
library(knitr)
library(readr)

opts_chunk$set("tidy" = TRUE)
opts_chunk$set("echo" = FALSE)
opts_chunk$set("eval" = TRUE)
opts_chunk$set("warning" = FALSE)
opts_chunk$set("cache" = FALSE)
opts_chunk$set("message" = FALSE)

```

```{r statistics}
#metadata

# model performance
l2_performance <- '../data/process/l2_otu/combined_best_hp_results_L2_Logistic_Regression.csv'

l2_performance_df <- read_csv(l2_performance)

temp_data <- summary(l2_performance_df$test_aucs)
model_perf_median <- format(round(temp_data[['Median']], 3), nsmall=3)
model_perf_25 <- format(round(temp_data[['1st Qu.']], 3), nsmall=3)
model_perf_75 <- format(round(temp_data[['3rd Qu.']], 3), nsmall=3)
rm(l2_performance_df, temp_data, l2_performance)

```


\vspace{30mm}


Running title: Clearance of *Clostridioides difficile* colonization is associated with antibiotic-specific bacterial changes

\vspace{20mm}

Nicholas A. Lesniak$^1$, Alyxandria M. Schubert$^{1,2}$, Hamide Sinani$^1$, Patrick D. Schloss$^{1,\dagger}$

\vspace{30mm}

$\dagger$ To whom correspondence should be addressed: pschloss@umich.edu

1\. Department of Microbiology and Immunology, University of Michigan, Ann Arbor, MI
2\. Current address: Laboratory of Mucosal Pathogens and Cellular Immunology, Division of Bacterial Pathogens and Allergenic Products, Office of Vaccines Research and Review, Center for Biologics Evaluation and Research, United States Food and Drug Administration, Silver Spring, Maryland, United States of America.




\newpage
\linenumbers


## Abstract

The gut bacterial community prevents pathogens from colonizing the intestine. Previous studies have associated specific bacteria with eliminating *Clostridioides difficile* colonization across different community perturbations. However, those bacteria alone have been unable to eliminate *C. difficile* colonization. In order to elucidate the changes necessary to eliminate colonization, we investigated how changes in bacterial abundance compare between communities able and unable to clear *C. difficile* colonization. We treated mice with titrated doses of antibiotics prior to *C. difficile* challenge which resulted in no colonization, clearance of colonization, or persistent colonization. Previously, we observed clindamycin-treated mice were susceptible to colonization but spontaneously cleared the colonization. Therefore, we investigated if other antibiotic-treated mice could be colonized and spontaneously clear the colonization. We found reduced doses of cefoperazone and streptomycin permitted colonization and then clearance of *C. difficile*. In communities able to clear colonization, we found each antibiotic had specific community changes and predicted interactions with *C. difficile*. Clindamycin treatment led to a bloom in populations related to *Enterobacteriaceae* and clearance of *C. difficile* was concurrent with the reduction of those blooming populations and the restoration of community members related to the *Porphyromonadaceae* and *Bacteroides*. Cefoperazone created a susceptible community characterized by a drastic reduction in the community diversity, interactions, and a sustained increase in abundance of many facultative anaerobes, including *Enterobacteriaceae*. Lastly, clearance in streptomycin-treated mice was associated with the recovery of multiple members of the *Porphyromonadaceae*, with little overlap in OTUs observed in the clindamycin treatment. Further elucidation of how *C. difficile* colonization is eliminated from different gut bacterial communities will improve *C. difficile* infection treatments.

\newpage
## Importance

The community of microorganisms, known as the microbiota, in our intestines prevents pathogens, such as *C. difficile*, from establishing themselves and causing infection. This is known as colonization resistance. However, when a person takes antibiotics, their gut microbiota is disturbed. This disruption allows *C. difficile* to colonize. *C. difficile* infections (CDI) are primarily treated with antibiotics, which frequently leads to recurrent infections because the microbiota have not yet returned to a resistant state. The infection cycle ends when the fecal microbiota from a resistant person are transplanted into the susceptible person. Although this treatment is highly effective, we do not understand the mechanism of resistance. We hope to improve the treatment of CDI through elucidating how the bacterial community eliminates *C. difficile* colonization. We found *C. difficile* was able to colonize susceptible mice but was spontaneously eliminated in an antibiotic-treatment specific manner. These data indicate each community had different requirements for clearing colonization. Understanding how different communities clear colonization will reveal targets to improve CDI treatments.




\newpage


## Introduction

A complex consortium of bacteria and microbes that inhabits our gut, known as the microbiota, prevent pathogens from colonizing and causing disease. This protection, known as colonization resistance, is mediated through many mechanisms such as activating host immune responses, competing for nutrients, producing antimicrobials, and contributing to the maintenance of the mucosal barrier [@ducarmon2019]. However, perturbations to the intestinal community or these functions opens the possibility that a pathogen can colonize [@britton2012]. For example, the use of antibiotics perturb the gut microbiota and can lead to *Clostridioides difficile* infection (CDI).

CDI is especially problematic due to its burden on the healthcare system [@lessa2015; @zimlichman2013]. *C. difficile* can cause severe disease, such as toxic megacolon, nosocomial diarrhea, and death [@spigaglia2016]. CDI is primarily treated with antibiotics [@dieterle2019]. CDIs recalcitrant to antibiotics are eliminated by restoring the community with a fecal microbiota transplant (FMT), returning the perturbed community to a healthier protective state [@juul2018; @seekatz2014]. However, FMTs are not always effective against CDI and have the risk of transferring a secondary infection [@patron2017; @defilipp2019]. Therefore, we need to better understand how the microbiota clears the infection to develop more effective treatments.

Previous research has shown that the microbiota affects *C. difficile* colonization. Mouse models have identified potential mechanisms of colonization resistance such as bile salt metabolism and nutrient competition [@buffie2015; @fletcher2018; @reed2020;  @jenior2017]. However, studies that have restored those functions were unable to restore complete resistance [@lawley2012; @mcdonald2018]. This could be attributed to the complexity of the community and the mechanisms of colonization resistance [@ghimire2019; @auchtung2020]. We previously showed that when *C. difficile* colonizes different antibiotic-treated murine communities it modifies its metabolism to fit each specific environment [@schubert2015; @jenior2017; @jenior2018]. Therefore, we investigated the bacterial community dynamics concurrent with *C. difficile* elimination across uniquely perturbed communities.

Jenior et al. observed that clindamycin-treated mice spontaneously cleared *C. difficile* colonization whereas mice treated with cefoperazone and streptomycin did not [@jenior2018]. Here, we continued to explore the different effects these three antibiotics have on *C. difficile* colonization. The purpose of this study was to elucidate the gut bacterial community changes concurrent with elimination of *C. difficile* colonization. We hypothesized that each colonized community has perturbation-specific susceptibilities and requires specific changes to clear the pathogen. To induce a less severe perturbation, we reduced the doses of cefoperazone and streptomycin. This resulted in communities that were initially colonized to a high level (>$10^{6}$) and then spontaneously cleared. We found each antibiotic resulted in unique changes in the microbiota that were associated with the persistence or clearance of *C. difficile*. These data further support the hypothesis that *C. difficile* can exploit numerous niches in perturbed communities.



## Results

**Reduced doses of cefoperazone and streptomycin allowed communities to spontaneously clear *C. difficile* colonization.**  To understand the dynamics of colonization and clearance of *C. difficile*, we first identified conditions which would allow colonization and clearance. Beginning with clindamycin, mice were treated with intraperitoneal injection of clindamycin (10 mg/kg) one day prior to challenge with *C. difficile*. All mice were colonized to a high level (median CFU = $3.07 \times 10^7$) the next day and cleared the colonization within 10 days; 6 mice cleared *C. difficile* within 6 days (Figure 1A). Previous *C. difficile* infection models using cefoperazone and streptomycin have not demonstrated clearance. So we next explored whether cefoperazone and streptomycin could permit colonization and subsequent clearance with lower doses. We began with replicating the previously established *C. difficile* infection models for these antibiotics [@jenior2018]. We treated mice with cefoperazone or streptomycin in their drinking water (0.5 mg/mL and 5 mg/mL, respectively) and then challenged them with *C. difficile*. For both antibiotics, *C. difficile* colonization was maintained for the duration of the experiment as previously demonstrated (Figure 1B-C) [@jenior2018]. Then we repeated the *C. difficile* challenge with reduced doses of the antibiotics (cefoperazone - 0.3 and 0.1 mg/mL; streptomycin - 0.5 and 0.1 mg/mL ). For both antibiotic treatments, the lowest dose resulted in either no colonization (N = 8) or a transient, low level colonization (n = 8, median length = 1 day, median CFU = $2.8 \times 10^3$) (Figure 1B-C). The intermediate dose of both antibiotics resulted in a high level colonization (median CFU = $3.5 \times 10^6$) and half (N = 8 of 16) of the mice clearing the colonization within 10 days. Based on our previous research, which showed each of these antibiotics uniquely changed the microbiota, we hypothesize that the microbiota varied across these antibiotic treatments that resulted in colonization clearance.

**Clearance of *C. difficile* has associated with antibiotic-specific changes to the microbiota community.** We analyzed the 16S rRNA gene sequences to identify the community features related to *C. difficile* colonization and clearance. First, we compared the most abundant bacterial genera of the communities at the time of *C. difficile* challenge. The clindamycin-treated mice became dominated by relatives of *Enterobacteriaceae* (Figure 1D). Increasing the dose of cefoperazone shifted the dominant community members from a community dominated by relatives of the *Porphyromonadaceae*, *Bacteroides* and *Akkermansia* to one dominated by relatives of the *Lactobacillus* and *Escherichia* with an increase in the non-dominant populations (Figure 1E). Increasing the dose of streptomycin  maintained the abundance of relatives of the *Porphyromonadaceae* and *Bacteroides*, but reduced most of the other genera including populations of the *Lactobacillus*, *Lachnospiraceae*, *Ruminococcaceae*, *Alistipes*, and *Clostridiales* (Figure 1F). Comparing across antibiotics, the differences int the microbiota at the time of *C. difficile* challenge support the hypothesis that the antibiotics have different effects on the community but it is unclear if those differences are related to the *C. difficile* clearance. Therefore, we began investigating the changes in microbiota diversity relative to whether *C. difficile* remained colonized. Clindamycin treatment decreased the alpha diversity and similarity to the pre-clindamycin community but did not restore the community diversity or similarity to the pre-clindamycin microbiota within the time period that *C. difficile* was cleared (Figure 2A). Cefoperazone treatment resulted in a range of changes to the community structure, from close resemblance to structurally distinct. For the communities that cleared *C.difficile*, diversity was maintained through antibiotic treatment and at the end of the experiment with only slight decrease in alpha diversity and community similarity with anitibiotic treatment. The mice treated with cefoperazone that remained colonized experienced an increase in alpha diversity, likely driven by the decrease in highly abundant populations and increase in low abundant populations (Figure 1E). However, those changes in alpha diversity were no longer present at the end of the experiment (Figure 3B). The colonized cefoperazone-treated communites had a large persistent shift in community structure with the antibiotic treatment which remained through the end of the experiment (Figure 3B). Streptomycin-treated mice became mildly dissimilar and less diverse with streptomycin treatment but returned to closely resemble the pre-antibiotic community at the end of the experiment, with a slightly larger effect on the communities that remained colonized (Figure 3C). Finally, we investigated if communities that cleared *C. difficile* become more similar to each other. We compared the end point communities to the other end point communities within treatments and across treatments. End point community structures were more similar to those receiving the same antibiotic treatment than those treated with a different antibiotic, with the exception of colonized cefoperazone-treated mice (Figure S1). The endpoint community of persistently colonized cefoperazone-treated mice were equally dissimilar to other end point communities within and across antibiotic treatments but were more dissimilar to each other than the cefoperazone-treated mice that cleared colonization. Overall, these data suggest that cefoperazone-treated mice cleared *C.difficile* as its community structure returned to pre-antibiotic treatment structure, but the community did not fully return to its pre-antibiotic state for the other antibiotic treatment groups.

**OTU abundance changes associated with *C. difficile* colonization were unique to each antibiotic treatment.** Since the communities that were most similar were from the same antibiotic treatment, we investigated the microbiota changes within each treatment across time and for each outcome. First, we identified the differences between mice that cleared *C. difficile* colonization and mice that remained colonized at their initial pre-antibiotic state, at the time of challenge, and at the end of the experiment. At the time of challenge, cefoperazone-treated mice that remained colonized throughout the experiment were significantly enriched in facultative anaerobic populations of *Enterococcus*, *Pseudomonas*, *Staphylococcus* and *Enterobacteriaceae* (Figure 3A). At the end of the experiment, cefoperazone-treated mice, which cleared colonization, had significant enrichment in 10 different operational taxonomic units (OTUs) related to the *Porphyromonadaceae*. Streptomycin-treated mice had fewer population differences between the colonization outcomes. At time of challenge, streptomycin-treated mice that remained colonized had reductions in 2 OTUs related to the *Porphyromonadaceae* and were depleted of relatives of *Alistipes*, *Anaeroplasma*, and *Porphyromonadaceae* (Figure 3B). At the end of the experiment, streptomycin-treated mice that remained colonized were depleted of populations of *Turicibacter*, *Alistipes*, and *Lactobacillus*. Next, we identified the significant temporal changes that occurred between pre-antibiotic treatment, the time of challenge, and the end of the experiment. Clindamycin-treated mice had large decreases in 21 OTUs and a bloom of relatives of *Enterobacteriaceae* (Figure 4A). With the elimination of *C. difficile*, we observed a drastic reduction of the relatives of *Enterobacteriaceae* and recovery of 10 populations related to *Porphyromonadaceae*, *Bacteroides*, *Akkermansia*, *Lactobacillus*, *Bifidobacterium*, *Lachnospiraceae*, and *Clostridiales* (Figure 4A). The majority of significant differences in population abundances in cefoperazone-treated mice were in persistently colonized communities. These communities had a loss of numerous relatives of the *Porphyromonadaceae* and increases in the facultative anaerobes differentiating the cefoperazone-treated mice by outcome (Figure 4C, S2). Lastly, the temporal changes in streptomycin-treated mice were more subtle with fewer changes than the other treatments. At the time of challenge, the communities that cleared colonization had more *Porphyromonadaceae* than those that did not (Figure 4B, D). At the end of the experiment, all communities experienced recovery of the abundance of many of the populations changed by the cefoperazone treatment, but the communities that remained colonized had OTUs of *Alistipes*, *Lactobacillus*, and *Porphyromonadaceae* that remained reduced. Together, these data revealed that while there were commonly affected families, such as *Porphyromonadaceae*, *C. difficile* clearance associated with OTU changes specific to the antibiotic.

**Distinct features of the bacterial community at the time of infection predict end point colonization.** To determine whether the community composition at the time of *C. difficile* challenge could predict *C. difficile* clearance, we built a machine learning model using L~2~ Logistic Regression. We evaluated the predictive performance of the model using the area under the receiver operating characteristic curve (AUROC), where a value of 0.5 indicated the model is random and 1.0 indicated the model always correctly predicts the outcome. Our model resulted in a AUROC of `r model_perf_median` [IQR `r model_perf_25`-`r model_perf_75`], which suggested the model was able to use the relative abundance of OTUs at the time of challenge to accurately predict colonization clearance (Figure S3). To assess the important features, we randomly permuted each OTU feature by removing it from the training set to determine its effect on the prediction (Figure 5A). The most important feature was *Enterobacteriaceae*, which predicted clearance, appears to be strongly driven by the clindamycin data (Figure 5B, C). The remaining OTU features did not have a large effect on the model performance, which suggested to us that most of the model decision was spread across many features. These results revealed the model used the relative abundance data of the community members and the relationship between those abundances to correctly classify clearance. There were many OTUs with treatment and outcome specific abundance patterns that did not agree with the magnitude of the effect that OTU had on odds ratio of the model. For example, *Enterobacteriaceae* abundance influences the model to predict clearance (Figure 5B), however in experiments that used cefoperazone, the communities that remained colonized had higher abundances of *Enterobacteriaceae* than the communities that cleared colonization (Figure 5C). The model would have been corrected by the influence of other OTUs to correctly classify those samples. Therefore, the model used a different combination of multiple OTUs and relative abundances across treatments to predict *C. difficile* clearance. These data can offer a basis for hypotheses regarding the distinct combinations of bacteria that promote *C. difficile* clearance.

**Conditional independence networks reveal treatment-specific relationships between the community members and *C. difficile* during colonization clearance.** We next investigated the relationship between temporal changes in  the community and *C. difficile* by building a conditional independence network for each treatment using SPIEC-EASI (sparse inverse covariance estimation for ecological association inference) [@kurtz2015].  First, we focused on the first-order associations of *C. difficile* (Figure 6A). In clindamycin-treated mice, *C. difficile* had positive associations with relatives of *Enterobacteriaceae*, *Pseudomonas*, and *Olsenella* and negative associations with relatives of the *Lachnospiraceae* and *Clostridium* XIVa. *C. difficile* had limited associations in cefoperazone-treated mice; the primary association was positive with relatives of *Enterobacteriaceae*. In streptomycin-treated mice, *C. difficile* had negative associations with relatives of the *Porphyromonadaceae* and positive associations with populations of the *Ruminococcaceae*, *Bacteroidetes*, *Clostridium* IV and *Olsenella*. Next, we quantified degree centrality, the number of associations between each OTU for the whole network of each antibiotic and outcome, and betweenness centrality, the number of associations connecting two OTUs that pass through an OTU (Figure 6B). This analysis revealed cefoperazone treatment resulted in networks composed of primarily singular associations with much lower degree centrality (*P* < 0.05) and betweenness centrality (*P* < 0.05) than the other antibiotic treatments. Communities that were treated with cefoperazone that resulted in cleared or persistent colonization had 10 to 100-fold lower betweenness centrality values than communities treated with clindamycin or streptomycin. Collectively, these networks suggest *C. difficile* colonization was affected by unique sets OTUs for mice treated with clindamycin and streptomycin, but cefoperazone treatment eliminated bacteria critical to maintain the community interactions and had few populations that associated with *C. difficile*.



## Discussion




Previous studies have identified microbiota features associated *C. difficile* colonization resistance in either a set of closely related murine communities or collectively across many different susceptible communities [@lawley2012; @mcdonald2018; @reeves2011]. However, experiments that supplemented those features to susceptible communities were not able to eliminate colonization [@lawley2012; @mcdonald2018; @reeves2012]. Here, we explored within uniquely susceptible communities the community changes that associated with *C. difficile* elimination. Even though these communities all began with closely related communities, colonization clearance was not dependent on a return to their baseline community and occurred with antibiotic-specific changes. These data suggest that the same treatment to restore colonization resistance may not be effective in all cases. Further investigation into specific cases of *C. difficile* colonization will elucidate the niches *C. difficile* utilizes and inform interventions to eliminate the colonization.

In this study we have shown antibiotic treatment resulted in specific changes for both *C. difficile* colonization and its clearance. Clindamycin-treated mice were susceptible with a dominant bloom in *Enterobacteriaceae*-affiliated populations which naturally resolved. Clearance was associated with the resolving bloom and recovery of bacteria reduced by the antibiotic treatment. Therefore, the clindamycin-treated mice may naturally restore colonization resistance within a short time period. A greater delay between clindamycin treatment and *C. difficile* challenge may allow sufficient recovery of the microbiota to prevent colonization. Streptomycin-treated mice were susceptible with minor changes in diversity and few OTUs with relative abundance changes or differences between colonization outcomes. It is plausible in this scenario susceptibility is driven by the loss of a few critical bacteria. Many of the significantly different bacteria associated with colonization were members of the *Porphyromonadaceae* and communities that cleared colonization maintained more populations related to the *Porphyromonadaceae*. This observation led us to the hypothesis that multiple populations related to the *Porphyromonadaceae* are necessary to provide colonization resistance. Colonization resistance could potentially be restored through returning the diversity of the *Porphyromonadaceae* family. Cefoperazone-treated mice became susceptible with the expansion of numerous facultative anaerobes while suppressing many of the initially most abundant genera. Communities that cleared colonization were able to return to resemble their initial community and maintained a better connected network. With cefoperazone, the persistently colonized community appeared to have been inhibited from clearing the colonization due to the sustained presence of the facultative anaerobes and inability to recovery from the initial antibiotic perturbation. Thus, each antibiotic treatment resulted in different bacterial changes associated with the ability to eliminate *C. difficile*.

While we have described antibiotic treatment-specific changes, our observations are not expected to directly transfer to human studies [@hugenholtz2017; @wrzosek2018]. Murine gut communities differ by their care facility and even greater from human gut communities [@nagpal2018; @osbelt2020; @velazquez2019]. Therefore, we looked to understand the general trends as opposed to the experiment-specific events. We categorized the general changes, microbial relationships and potential interventions of these colonization scenarios into three themes. (i) Limited opportunity - a transient dominance by a facultative anaerobe permits *C. difficile* colonization but quickly recovers and *C. difficile* is not able to persist, as with clindamycin treatment. We hypothesize this susceptibility is due to a transient repression of the community and interventions may make the infection worse. If time alone is sufficient for the community to recover and eliminate infection, treating the community with an antibiotic or the preparation for an FMT may create new susceptibilities, eliminate protective functions or open new nutrient niches. This unnecessary intervention could lead to increased susceptibility, allowing *C. difficile* to spread to other niches and prolong the infection. This susceptibility may be identified by a disappearing facultative anaerobe bloom. (ii) Specific opportunity - a perturbation only affects a select portion of the microbiota, leading to minor relative abundance changes and a slight decrease in diversity, opening a limited specific niche for *C. difficile* to colonize, as with streptomycin treatment. We hypothesize a few specific bacteria would be necessary to recolonize the exposed niche and eliminate *C. difficile* colonization. A fecal microbiota transplant may seed the necessary bacteria to clear *C. difficile* from the exposed niche of the colonized community. (iii) Extensive opportunity - a significant perturbation occurs leading to a persistent increase in facultative anaerobes and many different niches become available, as with cefoperazone treatment. These communities appear to have been severely depleted and are likely lacking numerous protective functions. We hypothesize many niches become exposed and available for *C. difficile* to colonize. In this scenario, a full FMT may be insufficient to providing adequate diversity to recover all the susceptibilities. It may be necessary to do multiple FMTs or transplant an enriched fecal community to recover the microbiota enough to outcompete *C. difficile* for nutrient niches and replace missing protective functions. Analyzing each of these colonization scenarios individually reveals how uniquely each recovery happens.

Based on these observations, future investigations can further elucidate the uniqueness of susceptible communities and the requirements to restore colonization resistance. One common theme for susceptibility across treatments was increased facultative anaerobe abundance. This increase could be investigated for its ability to predict community susceptibility to *C. difficile* colonization. The facultative anaerobe blooms could be attributed  to the loss of oxygen level suppression by obligate anaerobes with antibiotic treatment [@winter2013; @riverachavez2017]. However, it is unclear what prevents the succession from facultative anaerobes back to obligate anaerobes, such as with cefoperazone-treated microbiotas. Future studies should investigate the relationship between facultative anaerobe blooms and susceptibility to colonization and interventions to recover the obligate anaerobes. Additionally, while we only used one *C. difficile* strain, it is possible due to strain differences, others may have different genetic capabilities and community interactions [@carlson2013]. For example, more virulent strains, like *C. difficile* VPI 10463, may affect the gut environment more since it produces more toxin [@lawley2012; @rao2015]. Those differences could impact the susceptible community and change the requirements to clear *C. difficile*. While we have shown the changes concurrent with *C. difficile* clearance are unique to each antibiotic and we have previously shown community functions at peak infection are different, it is unclear how the community functions contributing to *C. difficile* clearance compare across antibiotics [@jenior2018]. Examining the changes in transcription and metabolites during clearance will help define the necessary changes to eliminate *C. difficile*. Building upon the information presented in this study will elucidate how the microbiota can eliminate *C. difficile* colonization.

Here we have shown murine gut communities become susceptible to *C. difficile* colonization after three different antibiotic treatments and then subsequently clear the colonization. This set of experiments have shown each antibiotic treatment resulted in different community changes concurrent with *C. difficile* clearance. This difference suggests that one treatment for all *C. difficile* infections may not be appropriate. While our current use of FMT to eliminate recurrent CDI is highly effective, it does not work in all patients and has even resulted in adverse consequences [@juul2018; @seekatz2014; @patron2017; @defilipp2019]. The findings in this study may help explain why FMTs may be ineffective. While the FMT is transplanting a whole community, it may not be sufficiently replacing community members or functions to inhibit *C. difficile* or the FMT procedure itself maybe disrupting the natural recovery of colonization resistance. Additionally, this study demonstrates a method to improve our understanding of how the bacterial community responds to *C. difficile* colonization. The knowledge of how a community clears *C. difficile* colonization will advance our ability to develop targeted therapies to manage CDI.



## Materials and Methods

**Animal care.** All mice were obtained from a single breeding colony and maintained in specific-pathogen-free (SPF) conditions at the University of Michigan animal facility. All mouse protocols and experiments were approved by the University Committee on Use and Care of Animals at the University of Michigan and completed in agreement with approved guidelines.

**Antibiotic administration.** Mice were given one of three antibiotics, cefoperazone, clindamycin, or streptomycin. Cefoperazone (0.5, 0.3, or 0.1 mg/ml) and streptomycin (5, 0.5, or 0.1 mg/ml) were delivered via drinking water for 5 days. Clindamycin (10 mg/kg) was administered through intraperitoneal injection.

** *C. difficile* challenge.** Mice were returned to untreated drinking water for 24 hours before challenging with *C. difficile* strain 630$\Delta$erm spores. *C. difficile* spores were aliquoted from a single spore stock stored at 4$^\circ$C. Spore concentration was determined one week prior to the day of challenge [@sorg2009]. $10^{3}$ *C. difficile* spores were orally gavaged into each mouse. Once the gavages were completed, the remaining spore solution was serial diluted and plated to confirm accurate spore concentration was delivered.

**Sample collection.** Fecal samples were collected on the day antibiotic treatment was started, on the day of *C. difficile* challenge and the following 10 days. For the day of challenge and beyond, a fecal sample was also collected and  weighed. Then under anaerobic conditions a fecal sample was serial diluted in anaerobic phosphate-buffered saline and plated on TCCFA plates. After 24 hours of anaerobic incubation at 37$^\circ$C CFU were enumerated [@winston2016].

**DNA sequencing.** Total bacterial DNA was extracted from each fecal sample using MOBIO PowerSoil-htp 96-well soil DNA isolation kit. We created amplicons of the 16s rRNA gene V4 region and sequenced them using an Illumina MiSeq as described previously [@kozich2013].

**Sequence curation.** Sequences were processed using mothur(v.1.43.0) as previously described [@kozich2013, @schloss2009]. Briefly, we used a 3% dissimilarity cutoff to group sequences into operational taxonomic units (OTUs). We used a naive Bayesian classifier with the Ribosomal Database Project training set (version 16) to assign taxonomic classifications to each OTU [@wang2007]. With the fecal samples, we also sequenced a mock community with a known community composition and their true 16s rRNA gene sequences. We processed this mock community along with our samples to determine our sequence curation resulted in an error rate of 0.012%.

**Statistical analysis and modeling.** Diversity comparisons were calculated in mothur. To compare within community diversity, we calculated Sobs and Inverse Simpson. To compare across communities, we calculated dissimilarity matrices based on metric of Yue and Clayton [@yue2005]. Samples were rarified to 1,200 sequences per sample within calculating the distance matrices to ensure no biases remained due to uneven sampling.  OTUs were subsampled to 1,200 counts per sample and remaining statistical analysis and data visualization was performed in R (v3.5.1) with the tidyverse package (v1.3.0). Pairwise OTU-to-OTU comparisons were calculated by pairwise Wilcoxon rank sum test and then P values were corrected for multiple comparisons with a Benjamini and Hochberg adjustment for a type I error rate of 0.05 [@benjamini1995]. Logistic regression models were constructed with OTUs from all day 0 samples  using half of the samples to train and the other half to test the model. The model was developed from the caret R package (v6.0-85) and previously developed machine learning pipeline [@topcuoglu2020]. Conditional independence networks were calculated using  SPIEC-EASI (sparse inverse covariance estimation for ecological association inference) methods from the SpiecEasi R package after optimizing lambda to 0.001 with a network stability between 0.045 and 0.05 (v1.0.7) [@kurtz2015]. Network centrality measures degree and betweenness were calculated on whole networks using functions from the igraph R package (v1.2.4.1).

**Code availability.** Scripts necessary to reproduce our analysis and this paper are available in an online repository (https://github.com/SchlossLab/Lesniak_Clearance_XXXX_2020).

**Sequence data accession number.** All 16s rRNA gene sequence data and associated metadata are available through the Sequence Read Archive via accession no. SXX#######.

## Acknowledgements

Thank you to Begüm Topçuoglu and Sarah Tomkovich for critical discussion in the development and execution of this project.
This work was supported by several grants from the National Institutes for Health **R01GM099514, R01HG005975, U19AI090871, and P30DK034933. Additionally, NAL was supported by the Molecular Mechanisms of Microbial Pathogenesis training grant (NIH T32 AI007528).**
The funding agencies had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.

\newpage

## References

<div id="refs"></div>

\newpage


![Figure 1](../results/figures/figure_1.jpg)

**Figure 1. Antibiotic-treated murine communities spontaneously cleared *C. difficile* colonization** (A-C) Daily CFU of *C. difficile* in fecal samples of mice treated with clindamycin, cefoperazone, or streptomycin from time of challenge (Day 0) through 10 days post infection (dpi). The bold line is the median CFU of the group and the transparent lines are the individual mice. (D-F) Relative abundance of twelve most abundant genera at the time of *C. difficile* challenge, all other genera grouped into Other. Each column is an individual mouse. Row above abundance graph indicates colonization outcome; white - uncolonized, gray - colonized and cleared, black - colonized through end of experiment.  Color intensity is log~10~-transformed mean percent relative abundance of each day. Limit of detection (LOD). (clindamycin - 10 mg/kg N =11; cefoperazone - 0.5 mg/mL N = 5, 0.3 mg/mL N = 9, 0.1 mg/mL N = 2; streptomycin - 5.0 mg/mL N = 8, 0.5 mg/mL N = 7, 0.1 mg/mL N = 7).

\hfill\break

![Figure 2](../results/figures/figure_2.jpg)

**Figure 2. Microbiotia community diversity showed antibiotic-specific trends in *C. difficle* colonization clearance** For clindamycin (A), cefoperazone (B), and streptomycin (C), microbiota $\alpha$-diversity (S~obs~ and Inverse Simpson) and $\beta$-diversity ($\theta$~YC~) were compared at the initial pre-antibiotic treatment state, time of *C. difficile* challenge (TOC), and end of the experiment. $\beta$-diversity ($\theta$~YC~) was compared between the initial pre-antibiotic treatment state relative and all other initial pre-antibiotic treatment communities treated with the same antibitoic, the initial community to the same community at the time of *C. difficile* challenge, and the initial community to the same community at end of the experiment. (clindamycin - cleared N = 11; cefoperazone - cleared N = 7, colonized N = 9; streptomycin - cleared N = 9, colonized N = 11). Points for communities no longer colonized with *C. difficile* are unfilled. Points for communities that were still colonized are filled in.

\hfill\break

![Figure 3](../results/figures/figure_3.jpg)

**Figure 3. Individual bacteria abundance differences between *C. difficile* colonization outcomes are unique to each treatment** For cefoperazone (A) and streptomycin (B), the difference in the relative abundance of OTUs that are significantly different between communities that were no longer colonized by *C. difficile* and those that remained samples colonized within each antibiotic treatment for each time point. Unfilled circles show samples that cleared *C. difficile* and filled circles show samples that remained colonized. Initial points - green; TOI points - blue; End points - red. Lines connect points by OTU within each comparison to show difference in medians. All OTUs plotted were statistically significant, p < 0.05  and calculated by Wilcoxon rank sum test with Benjamini-Hochberg correction. Bold points are median relative abundance and transparent points are relative abundance of individual mice. Limit of detection (LOD).

\hfill\break

![Figure 4](../results/figures/figure_4.jpg)

**Figure 4. Individual bacteria abundance changes associated with *C. difficile* colonization are unique to each treatment** For clindamycin (A), cefoperazone (C), and streptomycin (B, D), the difference in the relative abundance of OTUs that are significantly different communities between time points within each *C. difficile* colonization outcome for each antibiotic treatment. Initial points - green; TOI points - blue; End points - red. Arrows point in the direction of the temporal change of the relative abundance. Lines connect points by OTU within each comparison to show difference in medians. Bold OTUs were shared across comparisons. All OTUs plotted were statistically significant, p < 0.05  and calculated by Wilcoxon rank sum test with Benjamini-Hochberg correction. Bold points are median relative abundance and transparent points are relative abundance of individual mice. Limit of detection (LOD).

\hfill\break

![Figure 5](../results/figures/figure_5.jpg)

**Figure 5. Distinct features of the bacterial community at the time of infection can classify end point colonization** (A) L~2~ logistic regression model feature importances determined by the negative effect randomizing an individual feature has on the model performance. All OTUs affecting performance shown. Gray column shows performance range of final model with all features included. (B) Distibution of feature coefficients used in L~2~ logistic regression model. Negative coefficient indicated abundance predicted the community cleared colonization (red) and positive coefficients indicated abundance predicted *C. difficile* remained colonized (blue). Feature label and boxplot colored to match the median coefficient. (C) Relative abundance difference in features used by L~2~ logistic regression model faceted by antibiotic treatment. Red boxplot - antibiotic-treated mice that were no longer colonized; Blue boxplot - antibiotic-treated mice that remained colonized. Boxplot shows median - horizontal bar, quartile range - box, range of non-outlier data - whiskers, data greater than 1.5 times the interquartile range - points.

\hfill\break

![Figure 6](../results/figures/figure_6.jpg)
**Figure 6. Conditional independence networks reveal treatment-specific relationships between the community and *C. difficile* during colonization clearance** (A) SPIEC-EASI (sparse inverse covariance estimation for ecological association inference) networks showing conditionally independent first-order relationships between *C. difficile* and the community as *C. difficile* is cleared for the gut environment. Nodes sized by median relative abundance of OTU. Edges thickness is determined by the relationship strength and colored red for negative interactions and blue for positive interactions. Bold OTUs were shared across networks. (B) Network centrality measured with betweenness, how many paths between two OTUs pass through an individual, and degree, how many connections an OTU has. Colored by antibiotic (cefoperazone - blue; clindamycin - red; streptomycin - yellow) and boxplots are filled in gray for networks that remained colonized. Boxplot shows median - horizontal bar, quartile range - box, range of non-outlier data - whiskers, data greater than 1.5 times the interquartile range - points.

\hfill\break

![Figure S1](../results/figures/figure_S1.jpg)
**Figure S1. Dissimilarity of community structures at the end of the experiment** For each antibiotic, clindamycin (red), cefoperazone (blue), and streptomycin (yellow), end point communities were compared to all other communities treated with the same antibiotic ('Within Antibiotic') and then to all other communities not treated with the same antibiotic ('Across Antibiotic'). Points for communities no longer colonized with *C. difficile* are unfilled. Points for communities that were still colonized are filled in.

\hfill\break

![Figure S2](../results/figures/figure_S2.jpg)
**Figure S2. Temporally differing OTU for cefoperazone-treated mice that cleared *C. difficile* colonization** Distribution of relative abundance for only significantly different OTU between time points, initial (green) , time of infection (blue) and end of experiment (NA). Solid points are individual samples. Open circles are median relative abundance. Arrow shows the change over time.

\hfill\break


![Figure S3](../results/figures/figure_S3.jpg)
**Figure S3. Bacterial community at the time of infection can classify endpoint colonization** Classification performance of L~2~ logistic regression. Area under the receiver-operator curve for classifying if the community will remain colonized based on the OTUs present at the time of *C. difficile* infection (Day 0).
