---
title: '**Clearance of *Clostridioides difficile* colonization is associated with antibiotic-specific
  bacterial changes**'
csl: mbio.csl
fontsize: 11pt
output:
  pdf_document:
    includes:
      in_header: header.tex
    keep_tex: yes
  word_document: default
bibliography: references.bib
---


```{r knitr_settings, eval=TRUE, echo=FALSE, cache=FALSE, warning=FALSE, message=FALSE}
library(knitr)
library(readr)

opts_chunk$set("tidy" = TRUE)
opts_chunk$set("echo" = FALSE)
opts_chunk$set("eval" = TRUE)
opts_chunk$set("warning" = FALSE)
opts_chunk$set("cache" = FALSE)
opts_chunk$set("message" = FALSE)

```

```{r statistics}
#metadata

# model performance
l2_performance <- '../data/process/otu/combined_best_hp_results_L2_Logistic_Regression.csv'
l2_performance_df <- read_csv(l2_performance)

temp_data <- l2_performance_df$test_aucs
model_perf_median <- format(round(median(temp_data), 3), nsmall=3)
model_perf_25 <- format(round(median(temp_data) - IQR(temp_data), 3), nsmall=3)
model_perf_75 <- format(round(median(temp_data) + IQR(temp_data), 3), nsmall=3)
rm(l2_performance_df, temp_data, l2_performance)

```


\vspace{30mm}


Running title: Clearance of *Clostridioides difficile* colonization is associated with antibiotic-specific bacterial changes

\vspace{20mm}

Nicholas A. Lesniak${^1}$, Alyxandria M. Schubert${^{1,2}}$, Hamide Sinani${^1}$, Patrick D. Schloss^1$\dagger$^

\vspace{30mm}

$\dagger$ To whom correspondence should be addressed: pschloss@umich.edu

1\. Department of Microbiology and Immunology, University of Michigan, Ann Arbor, MI
2\. Current address: Laboratory of Mucosal Pathogens and Cellular Immunology, Division of Bacterial Pathogens and Allergenic Products, Office of Vaccines Research and Review, Center for Biologics Evaluation and Research, United States Food and Drug Administration, Silver Spring, Maryland, United States of America.




\newpage
\linenumbers


## Abstract

The gut bacterial community prevents pathogens from colonizing the intestine. Although common bacteria associated with eliminating *Clostridioides difficile* colonization across different community perturbations have been identified, those bacteria alone have been unable to eliminate *C. difficile* colonization. In order to elucidate the changes necessary to eliminate colonization, we investigated how community changes compare between those able and unable to clear *C. difficile* colonization. We treated mice with titrated doses of antibiotics prior to *C. difficile* challenge which resulted in no colonization, clearance of colonization or persistent colonization. Previously, we observed clindamycin-treated communities were susceptible to colonization but spontaneously cleared the colonization. Therefore, we investigated if other antibiotic-treated communities could be colonized and spontaneously clear the colonization. We found intermediate doses of cefoperazone and streptomycin permitted colonization and clearance of *C. difficile*. In communities able to clear colonization, we found each antibiotic had specific community changes and interactions with *C. difficile*. Clindamycin led to a susceptibility associated with a bloom in populations related to *Escherichia*/*Shigella*. Clearance of *C. difficile* was concurrent with the reduction of those blooming populations and the restoration of community members related to the *Porphyromonadaceae* and *Bacteroides*. Cefoperazone created a susceptible community characterized by a drastic reduction in the community structure and a sustained increase in abundance of many facultative anaerobes. Lastly, clearance in streptomycin-treated communities was associated with the recovery of multiple members of the *Porphyromonadaceae*. Further elucidation of how *C. difficile* colonization is eliminated from different gut bacterial communities will improve *C. difficile* infection treatments.

\newpage
## Importance

The community of microorganisms, known as the microbiota, in our intestines prevents pathogens from establishing themselves and causing infection. This is known as colonization resistance. *Clostridioides difficile* is only able to colonize when a healthy microbiota is perturbed. The gut microbes affected by the antibiotic are no longer able to prevent colonization, which permits *C. difficile* to colonize. *C. difficile* infections (CDI) are primarily treated with either antibiotics or transplanting a fecal community. These treatments are not always effective and have risks associated. We hope to improve our management of CDI through elucidating how the bacterial community eliminates *C. difficile* colonization. We found *C. difficile* was able to colonize but was spontaneously eliminated in an antibiotic-treatment specific manner. These data indicate each community had different requirements for clearing colonization. Understanding how different communities clear colonization will reveal targeted interventions to treat CDI.




\newpage


## Introduction

A complex consortium of bacteria and microbes that inhabits our gut, known as the microbiome, prevent pathogens from colonizing and causing disease. This protection, known as colonization resistance, is mediated through many mechanisms such as activating a host immune response, competing for nutrients, producing antimicrobials, and contributing to the maintenance of the mucosal barrier [@ducarmon2019]. Any perturbation to the intestinal community or these functions opens the possibility that a pathogen can colonize [@britton2012]. For example, the use of an antibiotic perturbs the gut microbiome and can lead to *Clostridioides difficile* infection (CDI).

CDI is especially problematic due to its burden on the healthcare system [@lessa2015; @zimlichman2013]. *C. difficile* can cause severe disease, such as toxic megacolon, nosocomial diarrhea, and death [@spigaglia2016]. CDI is primarily treated with antibiotics [@dieterle2019]. CDIs recalcitrant to antibiotics are eliminated by restoring the community with a fecal microbiota transplant (FMT), returning the perturbed community to a healthier protective state [@juul2018; @seekatz2014]. However, FMTs are not always effective against CDI and have the risk of transferring a secondary infection [@patron2017; @defilipp2019]. Therefore, we need to better understand how the microbiome clears the infection to develop better treatments.

To identify the susceptibilities of the microbial intestinal communities, we need to understand how the microbiome resists colonization. Previous research in mouse models has identified potential mechanisms of colonization resistance such as bile salt metabolism and nutrient competition [@buffie2015; @fletcher2018; @reed2020;  @jenior2017]. However, studies that have restored those functions were unable to restore complete resistance [@lawley2012; @mcdonald2018]. This could be attributed to the complexity of the community and the mechanisms of colonization resistance [@ghimire2019; @auchtung2020]. Previously, we have shown that when *C. difficile* colonizes different antibiotic-treated murine communities it modifies its metabolism to fit each specific environment [@schubert2015; @jenior2017; @jenior2018]. Therefore, we wanted to compare the bacterial community dynamics of *C. difficile* elimination across differently susceptible communities.

The purpose of this study was to elucidate the gut bacterial community changes concurrent with elimination of *C. difficile* colonization. We hypothesized each colonized community has perturbation-specific susceptibilities and requires specific changes to eliminate colonization. Jenior et al. made the observation that clindamycin-treated communities spontaneously cleared *C. difficile* colonization whereas mice treated with cefoperazone and streptomycin did not [@jenior2018]. Here, we continued to explore the different effects these three antibiotics have on *C. difficile* colonization. To induce a less severe perturbation, we reduced the doses of cefoperazone and streptomycin. This resulted in communities that were initially colonized to a high level (>$10^{6}$) and spontaneously cleared *C. difficile*. We asked whether changes to the microbiome were associated with clearance were unique to the antibiotic treatment. Using 16S rRNA gene sequencing, we observed that each community had antibiotic-specific changes that were associated with the persistence and clearance of *C. difficile*. These data further support the hypothesis that each case of CDI may need to be treated uniquely for the specific niche *C. difficile* exploits to establish an infection.



## Results

**Clindamycin-treated communities spontaneously cleared *C. difficile* colonization without recovering diversity.** In order to understand how the gut microbiota clear *C. difficile* colonization, we began by investigating the changes associated with clearance in clindamycin treated communities. Mice (n = 11) were treated with intraperitoneal injection of clindamycin (10 mg/kg) one day prior to challenge with *C. difficile*. All mice were colonized to a high level and cleared the colonization within 10 days; 50% of mice cleared *C. difficile* within 6 days (Figure 1A). We then analyzed the 16S rRNA gene sequences to understand the community changes occurring while *C. difficile* colonization was being eliminated. The diversity of the community decreased with antibiotic treatment, but did not recover within the time period that *C. difficile* colonization was cleared (Figure 1B). Interestingly, the changes in diversity in *C. difficile* challenged communities were similar to the changes observed in mock challenged communities. Next, we determined how the abundance of the most abundant genera changed relative to the abundance of *C. difficile*. On day 0, the community was dominated by relatives of *Escherichia*/*Shigella*, the abundance of these populations decreased over the following six days, concurrent with reduction in the abundance of *C. difficile* (Figure 1C). We observed an opposite trend in the initially dominant community members the *Porphyromonadaceae* and *Bacteroides*. In summary, the clindamycin-treated community allowed *C. difficile* colonization with a transient bloom of *Escherichia*/*Shigella* and cleared the colonization as it began to return to its initial community structure without completely returning.

** *C. difficile* is spontaneously cleared at reduced doses of cefoperazone and streptomycin.** We next determined whether cefoperazone and streptomycin could permit colonization and subsequent clearance if we used lower doses. First, we replicated the previously established *C. difficile* infection model [@jenior2018]. We treated mice with cefoperazone or streptomycin (0.5 mg/mL and 5 mg/mL, respectively) and then challenged them with *C. difficile*. These mice were followed for 10 days, sampling stool daily for *C. difficile* CFU and bacterial 16S rRNA gene sequences to observe long-term colonization dynamics. In both experiments, *C. difficile* colonization was maintained for the duration of the experiment as previously observed (Figure 2A) [@jenior2018]. Next, we repeated the *C. difficile* challenge with a two-step reduction in the doses of antibiotics (cefoperazone - 0.3 and 0.1 mg/mL; streptomycin - 0.5 and 0.1 mg/mL ). For both antibiotic treatments, we observed the lowest doses were either not colonized (n = 8) or colonized at a low level for a short time (n = 8, median length = 1 day, median CFU = 2800) (Figure 2B). The intermediate dose of both antibiotics resulted in a high level colonization (median CFU = $3.5 \times 10^6$) and resulted in half (n = 8 of 16) of the mice clearing the colonization. We next compared the 16S rRNA gene sequences of the communities at the time of infection. Increasing cefoperazone doses shifted the dominant community members from a community dominated by relatives of the *Porphyromonadaceae*, *Bacteroides* and *Akkermansia* to one dominated by relatives of the *Lactobacillus* and *Escherichia* (Figure 2C). Increased streptomycin doses maintained the abundance of relatives of the *Porphyromonadaceae*, *Bacteroides* and *Akkermansia*, but reduced most of the other genera including populations of the *Alistipes*, *Anaeroplasma*, *Lachnospiraceae*, *Lactobacillus*, and *Ruminococcaceae*. By modifying the dose of antibiotics, we were able to generate distinct communities capable of clearing *C. difficile*.

**Bacterial abundance changes associated with *C. difficile* colonization are unique to each antibiotic treatment.** With the observation spontaneous clearance could occur with each antibiotic, we compared the differences in communities by the colonization outcome and across the antibiotic treatments. First, we compared the microbiome structures at major time points, initial, time of infection (TOI) and the end of experiment, using the $\beta$-diversity $\theta$~YC~ dissimilarities. Clindamycin-treated communities had high $\theta$~YC~ values with antibiotic treatment (Initial vs TOI) and remained high through the end of the experiment (End vs Initial) relative to their initial community, suggesting the microbiome had a distinct structure through clearing the colonization (Figure 3A). Cefoperazone treatment resulted in a range in structural differences, from close resemblance to structurally distinct. $\theta$~YC~ values of the end point mirrored the colonization outcome (i.e., communities that remained dissimilar were the communities that remained colonized) (Figure 3B). Streptomycin treatment overall resulted in little differences in structure  between the two colonization outcomes. Streptomycin-treated communities became mildly dissimilar with streptomycin but returned to closely resemble the initial community at the end of the experiment (Figure 3C). Additionally, we investigated if communities that cleared the colonization were becoming more similar. We compared the end point communities within and across treatments. Community structures were more similar to those receiving the same antibiotic treatment than those treated with a different antibiotic, with the exception of colonized cefoperazone-treated communities (Figure S1). Cefoperazone-treated communities persistently colonized were more dissimilar at both the time of infection and end of experiment. Overall, these data suggest returning to the initial community is critical for cefoperazone to clear colonization, but is not necessary for the other antibiotic treated communities.

Next, we investigated the microbiome changes within each antibiotic treatment which occurred over time and resulted in either *C. difficile* colonization or clearance. First, we identified the differences between outcomes within each time point. At the time of infection, cefoperazone-treated communities that remained colonized were significantly enriched in facultative anaerobic populations of the *Streptococcus*, *Lactobacillus*, *Enterococcus*, *Pseudomonas*, *Staphylococcus* and *Escherichia*/*Shigella* (Figure 3D). Cefoperazone-treated communities which cleared colonization had significant enrichment in eight different populations related to the *Porphyromonadaceae* at the end of the experiment. Streptomycin-treated communities had fewer differences between outcomes. Communities that remained colonized were overall depleted of *Alistipes* and at time of infection were depleted of two populations related to the *Porphyromonadaceae* and one relative of *Anaeroplasma* (Figure 3E). Next, we determined the significant temporal changes occurring between major time points within each colonization outcome. Clindamycin-treated communities showed large decreases in 23 operational taxonomic units (OTUs) and a near complete dominant bloom of relatives of *Escherichia/Shigella*. With the elimination of *C. difficile*, we observed a drastic reduction of the *Escherichia*/*Shigella* relatives and recovery of 9 populations related to *Porphyromonadaceae*, *Bacteroides*, *Akkermansia*, *Lactobacillus*, *Bifidobacterium* and *Lachnospiraceae* (Figure 3F). The differences in cefoperazone-treated communities were primarily observed in persistently colonized communities. These communities had a loss of numerous relatives of the *Porphyromonadaceae* and increases in the facultative anaerobes differentiating the cefoperazone-treated communities by outcome (Figure 3H, S2). Lastly, the temporal changes in streptomycin-treated communities were more subtle with fewer community member changes than the other treatments. The communities no longer colonized had more populations of relatives of the *Porphyromonadaceae* increased and fewer depleted OTUs at the time of infection (Figure 3G, I). At the end of the experiment cleared communities had mulitple OTUs shift towards 1% relative abundance, but when *C. difficile* persisted relatives of *Olsenella* were depleted and three populations related to the *Porphyromonadaceae* increased to nearly 10% relative abundance. When looking at higher taxonomic levels, there was less overlap in significant bacterial groups across treatments and loss of the resolution of changes that occurred, such as populations related to the *Porphyromonadaceae* in streptomycin-treatment (Figure S3). Together, these data revealed there are some commonly affected genera, such as *Porphyromonadaceae*, but there were many antibiotic-treatment specific OTU changes associated with *C. difficile* clearance.

**Distinct features of the bacterial community at the time of infection predict end point colonization.** To determine if the community composition at the time of *C. difficile* challenge can classify the *C. difficile* clearance, we employed a machine learning model, L~2~ Logistic Regression. With relative abundance of OTUs in all day 0 samples, the model was able to correctly classify our samples with an AUROC of `r model_perf_median` [IQR `r model_perf_25`-`r model_perf_75`] (Figure S4). We expect this model to be overfit due to the strong correlation between antibiotic treatments and outcome. However, since our goal was to determine the features differentiating the outcome we did not remove treatment dependent effects. To assess the important features, we randomly permuted each OTU feature to determine its contribution to outcome classification (Figure 4A). These results revealed the model used the relative abundance data of a subset of the community members and the relationship between those abundances to correctly classify clearance (Figure 4B). While some OTUs had similar relationships across antibiotic treatments, like many populations related to *Lachnospiraceae*, there were many OTUs with treatment-specific abundance relationships. For example, high abundance of relatives of *Escherichia*/*Shigella* did not necessarily indicate one outcome, but including the abundance of other community members allowed for accurate outcome classification. Therefore, the model used OTUs with differential relative abundances across treatments from multiple OTUs to classify the *C. difficile* clearance. These data can offer a basis to hypothesize the distinct combinations of bacteria that promote *C. difficile* clearance.

**Conditional independence networks reveal treatment-specific relationships between the community members and *C. difficile* during colonization clearance.** Looking further into the temporal data beyond cross-sectional differences, we investigated the relationship between temporal changes in  the community and *C. difficile*. To understand how the changes in bacterial abundances directly related to changes in *C. difficile* abundance, we developed a conditional independence network for each treatment using SPIEC-EASI (sparse inverse covariance estimation for ecological association inference) [@kurtz2015].  First, we focused on the first-order connections of *C. difficile* (Figure 5A). In clindamycin-treated communities, *C. difficile* had positive relationships with relatives of *Escherichia*/*Shigella*, *Pseudomonas*, and *Olsenella* and negative relationships with populations related to the *Lachnospiraceae* and *Clostridium* XIVa. We observed *C. difficile* had limited interactions in cefoperazone-treated communities, the primary relationship was a positive with relatives of *Escherichia*/*Shigella*. In streptomycin-treated communities, *C. difficile* had negative relationships with relatives of the *Porphyromonadaceae* and positive relationships with populations of the *Ruminococcaceae*, *Bacteroidetes*, *Clostridium* IV and *Olsenella*. Finally, for the whole network of each antibiotic and outcome we determined the number of connections between each OTU, known as degree centrality, and the number of paths connecting two OTUs pass through an OTU, known as betweenness centrality (Figure 5B). This analysis revealed cefoperazone treatment resulted in networks with much lower degree centrality, composed of primarily singular connections. Also, those networks had reduced betweenness centrality; cleared and colonized communities were one and two logs lower ,respectively, than communities treated with clindamycin or streptomycin. Collectively, these networks suggest that unique subsets of bacteria interacted with *C. difficile* in each antibiotic-treated community and cefoperazone treatment eliminated bacteria critical to community wide interactions.



## Discussion



Previous studies have identified microbiome features associated *C. difficile* colonization resistance in either a set of closely related murine communities or collectively across many different susceptible communities [@lawley2012; @mcdonald2018; @reeves2011]. However, experiments supplemented those features to susceptible communities, none were able to eliminate colonization [@lawley2012; @mcdonald2018; @reeves2012]. Here, we explored community changes within uniquely susceptible communities that associated with *C. difficile* colonization. Even though these communities all began with closely related communities, colonization clearance occurred with antibiotic-specific changes and clearance was not dependent on a return to their baseline community. These data suggest that the same treatment to restore colonization resistance may not be effective in all cases. Further investigation into specific cases of *C. difficile* colonization will elucidate the niches *C. difficile* utilizes and inform interventions to eliminate the colonization.

In this study we have shown antibiotic treatment resulted in specific changes for both *C. difficile* colonization and its clearance. Clindamycin-treated communities were susceptible with a dominant bloom in *Escherichia*/*Shigella*-affiliated populations which naturally resolved. Clearance was associated with the resolving bloom and recovery of bacteria reduced by the antibiotic treatment. Interestingly, the community recovery was similar for *C. difficile*-challenged and mock-challenged communities. Therefore, the clindamycin-treated communities may naturally restore colonization resistance within a short time period. A greater delay between clindamycin treatment and *C. difficile* challenge may allow sufficient recovery of the microbiome to prevent colonization. Streptomycin-treated communities were susceptible with minor changes in diversity and few OTUs with relative abundance changes or differences between colonization outcomes. It is plausible in this scenario susceptibility is driven by the loss of a few critical bacteria. Many of the significantly different bacteria associated with colonization were members of the *Porphyromonadaceae* and communities that cleared colonization maintained more populations related to the *Porphyromonadaceae*. However, the abundance of this genus did not differ greatly. This conflict led us to the hypothesis that multiple populations related to the *Porphyromonadaceae* are necessary to provide colonization resistance. Colonization resistance could potentially be restored through returning the diversity populations from the *Porphyromonadaceae* family. Cefoperazone-treated communities became susceptible with the expansion of numerous facultative anaerobes while suppressing many of the initially most abundant genera. Communities that cleared colonization were able to return to near their original community and maintained a better connected network. In this treatment, the persistently colonized community appeared to have been inhibited from clearing the colonization due to the sustained presence of the facultative anaerobes and inability to recovery from the initial perturbation. Thus, each antibiotic treatment resulted in bacterial changes associated with the ability to eliminate *C. difficile*.

While we have described antibiotic treatment-specific changes, our observations are not expected to directly transfer to human studies [@hugenholtz2017; @wrzosek2018]. Murine gut communities differ by their care facility and even greater from human gut communities [@nagpal2018; @osbelt2020; @velazquez2019]. Therefore, we looked to understand the general trends as opposed to the experiment-specific events. We categorized the general changes, microbial relationships and potential interventions of these colonization scenarios into three themes. (i) Limited opportunity - a transient dominance by a facultative anaerobe permits *C. difficile* colonization but quickly recovers and *C. difficile* is not able to persist, as observed in clindamycin treatment. We hypothesize this susceptibility is due to a transient repression of the community and interventions could possibly make the infection worse. If time alone is sufficient for the community to recover and eliminate infection, treating the community with an antibiotic or the preparation for an FMT may create new susceptibilities, eliminate protective functions or open new nutrient niches. This unnecessary intervention could lead to increased susceptibility, allowing *C. difficile* to spread to other niches and prolong the infection. This susceptibility may be identified by a disappearing facultative anaerobe bloom. (ii) Specific opportunity - a perturbation only affects a select portion of the microbiome, leading to minor relative abundance changes and a slight decrease in diversity, opening a limited specific niche for *C. difficile* to colonize, as observed with streptomycin treatment. We hypothesize a few specific bacteria would be necessary to recolonize the exposed niche and eliminate *C. difficile* colonization. A fecal microbiota transplant may seed the necessary bacteria to clear *C. difficile* from the exposed niche of the colonized community. (iii) Extensive opportunity - a significant perturbation occurs leading to a persistent increase in facultative anaerobes and many different niches become available, as observed with cefoperazone treatment. These communities appear to have been severely depleted and are likely lacking numerous protective functions. We hypothesize many niches become exposed and available for *C. difficile* to colonize. In this scenario, a full FMT may be insufficient to providing sufficient diversity to recover all the susceptibilities. It may be necessary to do multiple FMTs or transplant an enriched fecal community to recover the community sufficient enough to outcompete *C. difficile* for nutrient niches and replace missing protective functions. Analyzing each of these colonization scenarios individually reveals how uniquely each recovery happens.

Based on these observations, future investigations can further elucidate the uniqueness of susceptible communities and the requirements to restore colonization resistance. One common theme for susceptibility across treatments was increased facultative anaerobe abundance. This increase could be investigated for its ability to predict community susceptibility to *C. difficile* colonization. The facultative anaerobe blooms could be attributed  to the loss of oxygen level suppression by obligate anaerobes with antibiotic treatment [@winter2013; @riverachavez2017]. However, it is unclear what prevents the succession of the primary community members from facultative anaerobes back to obligate anaerobes, such as with cefoperazone-treated microbiomes. Future studies should investigate the relationship between facultative anaerobe blooms and susceptibility to colonization and interventions to recover the obligate anaerobes. Additionally, while we only used one *C. difficile* strain, it is possible due to strain differences, others may have different genetic capabilities and community interactions [@carlson2013]. For example, more virulent strains, like *C. difficile* VPI 10463, may affect the gut environment more since it produces more toxin [@lawley2012; @rao2015]. Those differences could impact the susceptible community and change the requirements to clear *C. difficile*. While we have shown the changes concurrent with *C. difficile* clearance are unique to each antibiotic and we have previously shown community functions at peak infection are different, it is unclear how the community functions contributing to *C. difficile* clearance compare across antibiotics [@jenior2018]. Examining the changes in transcription and metabolites during clearance will help define the necessary changes to eliminate *C. difficile*. Building upon the information presented in this study will elucidate how the microbiome can eliminate *C. difficile* colonization.

Here we have shown murine gut communities become susceptible to *C. difficile* colonization after three different antibiotic treatments and then subsequently clear the colonization. This set of experiments have shown each antibiotic treatment resulted in different community changes concurrent with *C. difficile* clearance. This difference suggests that one treatment for all *C. difficile* infections may not be appropriate. While our current use of FMT to eliminate recurrent CDI is highly effective, it does not work in all patients and has even resulted in adverse consequences [@juul2018; @seekatz2014; @patron2017; @defilipp2019]. The findings in this study may help explain why FMTs may be ineffective. While the FMT is transplanting a whole community, it may not be sufficiently replacing community members or functions to inhibit *C. difficile* or the FMT procedure itself maybe disrupting the natural recovery of colonization resistance. Additionally, this study demonstrates a method to improve our understanding of how the bacterial community responds to *C. difficile* colonization. The knowledge of how a community clears *C. difficile* colonization will advance our ability to develop targeted therapies to manage CDI.



## Materials and Methods

**Animal care.** All mice were obtained from a single breeding colony and maintained in specific-pathogen-free (SPF) conditions at the University of Michigan animal facility. All mouse protocols and experiments were approved by the University Committee on Use and Care of Animals at the University of Michigan and completed in agreement with approved guidelines.

**Antibiotic administration.** Mice were given one of three antibiotics, cefoperazone, clindamycin, or streptomycin. Cefoperazone (0.5, 0.3, or 0.1 mg/ml) and streptomycin (5, 0.5, or 0.1 mg/ml) were delivered via drinking water for 5 days. Clindamycin (10 mg/kg) was administered through intraperitoneal injection.

** *C. difficile* challenge.** Mice were returned to untreated drinking water for 24 hours before challenging with *C. difficile* strain 630$\Delta$erm spores. *C. difficile* spores were aliquoted from a single spore stock stored at 4$^\circ$C. Spore concentration was determined one week prior to the day of challenge [@sorg2009]. $10^{3}$ *C. difficile* spores were orally gavaged into each mouse. Once the gavages were completed, the remaining spore solution was serial diluted and plated to confirm accurate spore concentration was delivered.

**Sample collection.** Fecal samples were collected on the day antibiotic treatment was started, on the day of *C. difficile* challenge and the following 10 days. For the day of challenge and beyond, a fecal sample was also collected and  weighed. Then under anaerobic conditions a fecal sample was serial diluted in anaerobic phosphate-buffered saline and plated on TCCFA plates. After 24 hours of anaerobic incubation at 37$^\circ$C CFU were enumerated [@winston2016].

**DNA sequencing.** Total bacterial DNA was extracted from each fecal sample using MOBIO PowerSoil-htp 96-well soil DNA isolation kit. We created amplicons of the 16s rRNA gene V4 region and sequenced them using an Illumina MiSeq as described previously [@kozich2013].

**Sequence curation.** Sequences were processed using mothur(v.1.XX) as previously described [@kozich2013, @schloss2009]. Briefly, we used a 3% dissimilarity cutoff to group sequences into operational taxonomic units (OTUs). We used a naive Bayesian classifier with the Ribosomal Database Project training set (version XX) to assign taxonomic classifications to each OTU [@wang2007]. With the fecal samples, we also sequenced a mock community with a known community composition and their true 16s rRNA gene sequences. We processed this mock community along with our samples to determine our sequence curation resulted in an error rate of X.XX%

**Statistical analysis and modeling.** Diversity comparisons were calculated in mothur. To compare within community diversity, we calculated Sobs and Inverse Simpson. To compare across communities, we calculated dissimilarity matrices based on metric of Yue and Clayton [@yue2005]. Samples were rarified to 2,000 sequences per sample within calculating the distance matrices to ensure no biases remained due to uneven sampling.  OTUs were subsampled to 2,000 counts per sample and remaining statistical analysis and data visualization was performed in R (v3.5.1) with the tidyverse package (v1.3.0). Pairwise OTU-to-OTU comparisons were calculated by pairwise Wilcoxon rank sum test and then P values were corrected for multiple comparisons with a Benjamini and Hochberg adjustment for a type I error rate of 0.05 [@benjamini1995]. Logistic regression models were constructed with OTUs from all day 0 samples  using half of the samples to train and the other half to test the model. The model was developed from the caret R package (v6.0-85) and previously developed machine learning pipeline [@topcuoglu2020]. Conditional independence networks were calculated using  SPIEC-EASI (sparse inverse covariance estimation for ecological association inference) methods from the SpiecEasi R package after optimizing lambda to 0.001 with a network stability between 0.045 and 0.05 (v1.0.7) [@kurtz2015]. Network centrality measures degree and betweenness were calculated on whole networks using functions from the igraph R package (v1.2.4.1).

**Code availability.** Scripts necessary to reproduce our analysis and this paper are available in an online repository (https://github.com/SchlossLab/Lesniak_Clearance_XXXX_2020).

**Sequence data accession number.** All 16s rRNA gene sequence data and associated metadata are available through the Sequence Read Archive via accession no. SXX#######.

## Acknowledgements

Thank you to Begüm Topçuoglu and Sarah Tomkovich for critical discussion in the development and execution of this project.
This work was supported by several grants from the National Institutes for Health **R01GM099514, R01HG005975, U19AI090871, and P30DK034933. Additionally, NAL was supported by the Molecular Mechanisms of Microbial Pathogenesis training grant (NIH T32 AI007528).**
The funding agencies had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.

\newpage

## References

<div id="refs"></div>

\newpage


![Figure 1](../results/figures/figure_1.jpg)

**Figure 1. Clindamycin-treated murine communities spontaneously clears *C. difficile* colonization** (A) Daily CFU of *C. difficile* in fecal samples of clindamycin-treated mice from time of challenge (Day 0) through 10 days post infection (dpi). The bold line is the median CFU of the group and the transparent lines are the individual mice.  (B) Quantification of diversity over time from pre-treatment (Day -1) through 10 dpi. Red colored boxes were challenged with *C. difficile* strain 630, *_WHEREAS_* gray boxes were mock-challenged. (C) Daily relative abundance of ten most abundant genera, all other genera grouped into Other. Color intensity is log~10~-transformed mean percent relative abundance of each day. (N = 11 mice). Limit of detection (LOD). Boxplot shows median - horizontal bar, quartile range - box, range of non-outlier data - whiskers, data greater than 1.5 times the interquartile range - points.

\hfill\break

![Figure 2](../results/figures/figure_2.jpg)

**Figure 2. *C. difficile* is spontaneously cleared when reducing cefoperazone and streptomycin concentrations** (A) Daily CFU of *C. difficile* in fecal samples of cefoperazone and streptomycin-treated mice from time of challenge (Day 0) through 10 dpi. Antibiotic concentrations used were *C. difficile* infection model levels (0.5 mg/mL and 5.0 mg/mL). (B) Daily CFU of *C. difficile* in fecal samples of cefoperazone and streptomycin-treated mice from time of challenge (Day 0) through 10 days post infection (dpi). Antibiotic concentrations used were two-step decreases from *C. difficile* infection model levels (0.3 and 0.1 mg/mL and 0.5 and 0.1 mg/mL, respectively). (C) Relative abundance of twelve most abundant genera on the day of *C. difficile* challenge (Day 0), all other genera grouped into Other. Color intensity is log10-transformed mean percent relative abundance of each day. (cefoperazone - 0.5 mg/mL N = 5, 0.3 mg/mL N = 9, 0.1 mg/mL N = 2; streptomycin - 5.0 mg/mL N = 8, 0.5 mg/mL N = 7, 0.1 mg/mL N = 7). Limit of detection (LOD). Bold line is the median CFU of the group and transparent lines are the individual mice.

\hfill\break

![Figure 3](../results/figures/figure_3.jpg)

**Figure 3. Individual bacteria abundance changes associated with *C. difficile* colonization are unique to each treatment** (A-C) $\beta$-diversity $\theta$~YC~ comparisons - Initial vs Initial compares one pre-antibiotic treatment to all other pre-antibiotic treatment samples; Initial vs Time of Infection (TOI) and End vs Initial compare time points within mouse, intra-group comparisons compare a sample to all other samples receiving the same antibiotic treatment and inter-group comparisons compares a sample to all other samples not receiving the same antibiotic treatment mouse to all other. Boxes colored by time point being compared (initial = green; time of infection (TOI) = blue; end = red) to the same time point or to the initial time point. Comparisons restricted to within antibiotic and outcome except inter-group comparisons. (D-E) The difference in the relative abundance of OTUs that are significantly different between communities that were no longer colonized by *C. difficile* and those that remained samples colonized within each antibiotic treatment for each time point. Unfilled circles show samples that cleared *C. difficile* and filled circles show samples that remained colonized. (F-I) The difference in the relative abundance of OTUs that are significantly different communities between time points within each *C. difficile* colonization outcome for each antibiotic treatment. Initial points - green; TOI points - blue; End points - red. Arrows point in the direction of the temporal change of the relative abundance. Lines connect points by OTU within each comparison to show difference in medians. Bold OTUs were shared across comparisons. All OTUs plotted in panels D-I were statistically significant, p < 0.05  and calculated by Wilcoxon rank sum test with Benjamini-Hochberg correction. Bold points are median relative abundance and transparent points are relative abundance of individual mice. Limit of detection (LOD). Boxplot shows median - horizontal bar, quartile range - box, range of non-outlier data - whiskers, data greater than 1.5 times the interquartile range - points.

\hfill\break

![Figure 4](../results/figures/figure_4.jpg)

**Figure 4. Distinct features of the bacterial community at the time of infection can classify end point colonization** (A) L~2~ logistic regression model feature importances determined by the negative effect randomizing an individual feature has on the model performance. All OTUs affecting performance shown. (B) Relative abundance difference in features used by L~2~ logistic regression model faceted by antibiotic treatment. Red boxplot - antibiotic-treated communities that were no longer colonized; Blue boxplot - antibiotic-treated communities that remained colonized. Boxplot shows median - horizontal bar, quartile range - box, range of non-outlier data - whiskers, data greater than 1.5 times the interquartile range - points.

\hfill\break

![Figure 5](../results/figures/figure_5.jpg)
**Figure 5. Conditional independence networks reveal treatment-specific relationships between the community and *C. difficile* during colonization clearance** (A) SPIEC-EASI (sparse inverse covariance estimation for ecological association inference) networks showing conditionally independent first-order relationships between *C. difficile* and the community as *C. difficile* is cleared for the gut environment. Nodes sized by median relative abundance of OTU. Edges thickness is determined by the relationship strength and colored red for negative interactions and blue for positive interactions. (B) Network centrality measured with betweenness, how many paths between two OTUs pass through an individual, and degree, how many connections an OTU has. Colored by antibiotic (cefoperazone - blue; clindamycin - red; streptomycin - yellow) and boxplots are filled in gray for networks that remained colonized. Boxplot shows median - horizontal bar, quartile range - box, range of non-outlier data - whiskers, data greater than 1.5 times the interquartile range - points.

\hfill\break

![Figure S1](../results/figures/figure_S1.jpg)
**Figure S1. Dissimilarity of community structures at the end of the experiment** For each antibiotic, clindamycin (red), cefoperazone (blue), and streptomycin (yellow), end point communities were compared to all other communities treated with the same antibiotic ('Within Antibiotic') and then to all other communities not treated with the same antibiotic ('Across Antibiotic'). Boxplots for communities no longer colonized with *C. difficile* are unfilled. The boxplots for communities that were still colonized are filled in (gray). Boxplot shows median - horizontal bar, quartile range - box, range of non-outlier data - whiskers, data greater than 1.5 times the interquartile range - points.

\hfill\break

![Figure S2](../results/figures/figure_S2.jpg)
**Figure S2. Temporally differing OTU for cefoperazone-treated communities that cleared *C. difficile* colonization** Distribution of relative abundance for only significantly different OTU between time points, initial (green) , time of infection (blue) and end of experiment (NA). Solid points are individual samples. Open circles are median relative abundance. Arrow shows the change over time.

\hfill\break

![Figure S3](../results/figures/figure_S3.jpg)
**Figure S3. Family-level abundance differences associated with *C. difficile* colonization are also unique but lose resolution of changes** (A) The difference in the relative abundance of families that are significantly different between communities that were no longer colonized by *C. difficile* and those that remained samples colonized within each antibiotic treatment for each time point. (B) The significantly different changes in the relative abundance of families between time points within each *C. difficile* colonization outcome for each antibiotic treatment. For data points unfilled outcome was clearance and those filled was colonized. Initial points - green; TOI points - blue; End points - red. Arrows point in the direction of the temporal change of the relative abundance. Lines connect points by family within each comparison to show difference in medians. Bold family was shared across comparisons. All families plotted were statistically significant, p < 0.05  and calculated by Wilcoxon rank sum test with Benjamini-Hochberg correction. Bold points are median relative abundance and transparent points are relative abundance of individual mice. Limit of detection (LOD). Boxplot shows median - horizontal bar, quartile range - box, range of non-outlier data - whiskers, data greater than 1.5 times the interquartile range - points.

\hfill\break

![Figure S4](../results/figures/figure_S4.jpg)
**Figure S4. Bacterial community at the time of infection can classify endpoint colonization** Classification performance of L~2~ logistic regression. Area under the receiver-operator curve for classifying if the community will remain colonized based on the OTUs present at the time of *C. difficile* infection (Day 0).
