---
title: '**Clearance of Clostridioides difficile colonization is associated with unique bacterial changes**'
csl: mbio.csl
fontsize: 12pt
output:
  word_document: default
  pdf_document:
    includes: null
    keep_tex: yes
  html_document: default
header-includes:
- \usepackage{setspace}
- \doublespacing
- \usepackage{lineno}
- \linenumbers
- \renewcommand{\familydefault}{\sfdefault}
- \usepackage{graphicx}
geometry: margin=1.0in
sansfont: Arial
bibliography: references.bib
---


```{r knitr_settings, eval=TRUE, echo=FALSE, cache=FALSE, warning=FALSE, message=FALSE}
library(knitr)
source('../code/functions.R')

opts_chunk$set("tidy" = TRUE)
opts_chunk$set("echo" = FALSE)
opts_chunk$set("eval" = TRUE)
opts_chunk$set("warning" = FALSE)
opts_chunk$set("cache" = FALSE)
opts_chunk$set("message" = FALSE)

```

```{r rdata_tables}
#metadata





```


\vspace{35mm}


Running title: Clearance of *Clostridioides difficile* colonization is associated with unique bacterial changes

\vspace{10mm}

Nicholas A. Lesniak${^1}$, Alyxandria M. Schubert${^2}$, Hamide Sinani${^3}$, Patrick D. Schloss${^1}$

\vspace{20mm}

$\dagger$ To whom correspondence should be addressed: pschloss@umich.edu

$1$ Department of Microbiology and Immunology, University of Michigan, Ann Arbor, MI 
$2$ FDA
$3$ ?????

\newpage
\linenumbers


## Abstract

The gut bacterial community plays an important role in preventing pathogens from colonizing the intestine. Many functions of the bacterial community preventing *Clostridioides difficile* from colonizing have been characterized, however we have not yet been able to successfully restore colonization resistance to a susceptible community. In order to elucidate the necessary changes for restoring colonization resistance, we sought to begin to understand how *C. difficile* is cleared from susceptible communities. We observed clindamycin treated communities were susceptible to *C. difficile* colonization and then spontaneously clear the colonization. Therefore, we investigated if other antibiotic-treated communities could become susceptible and subsequently spontaneously clear *C. difficile* colonization. We found intermediate dosages of cefoperazone and clindamycin permitted the colonization and subsequent clearance of *C. difficile*. While all treatments resulted in colonization and clearance of *C. difficile*, we found each antibiotic had unique communities, bacterial changes and interactions with *C. difficile*. Generally, we found three different trends in susceptibility - transient *Escherichia*/*Shigella* bloom, repression of a few non-dominant bacteria, and sustained presence of diverse facultative anaerobes. We found clindamycin treatment led to a susceptibility associated with a relative bloom in the facultative anaerobe, *Esherichia*/*Shigella*. The clearance of *C. difficile* was concurrent with the reduction of Escherichia/Shigella and the restoration of the dominant bacteria. Whereas, streptomycin lead to relatively few changes to the community and there were only a few bacteria changing and interacting with *C. difficile*. Clearance in this scenario was associated with multiple Porphyromonadaceae OTUs. Lastly, cefoperazone led to a drastic reduction in the community structure resulting in a sustained increase in abundance of many facultative anaerobes, allowing *C. difficile* to persist. Further elucidation of how *C. difficile* is colonizing different gut bacterial communities will help us to better identify and treat intestinal communities susceptible to *C. difficile* infection.

**Importance** The community of bacteria and other microbes in our intestines, our gut microbiome, provide us many benefits. One benefit of this bacterial community is preventing problematic bacteria from establishing itself in our intestine and causing an infection. This protection is known as colonization resistance. *Clostridioides difficile* is especially vunerable to colonization resistance. *C. difficile* is unable to colonize when a healthy microbiome is present. However, when the gut community is modified by, for example, taking antibiotics, the bacteria affected by the antibiotic are no longer able to prevent colonization. This perturbation opens opportunities for *C. difficile* to colonize. *C. difficile* infections are primarily treated with antibiotics, however antibiotics donâ€™t always work and can sometimes lead to recurrent infection. Therefore, we hope to understand how the bacterial community prevents and eliminates *C. difficile* colonization. We found that different bacterial communities colonized with *C. difficile* were able to spontaneously clear the colonization. Additionally, each treatment resulted in unique community changes with the clearance of *C. difficile*. These data indicate each community had different requirements for clearing the colonization. Understanding these clearance dynamics will better inform us the different ways *C. difficile* can persist and the microbiome can eliminate colonization.




\newpage

## Introduction

A complex consortium of bacteria and microbes inhabiting our gut prevent pathogens from colonizing and causing disease. This protection, known as colonization resistance, is mediated by bacteria through many mechanisms such as activating a host immune response, competing for nutrients, producing bacteriocins, and contributing to the maintenance of the mucosal barrier. Any perturbation of the intestinal community or these functions opens the possibility of a pathogen, like *Clostridioides difficile*, colonizing.

Once the gut community is perturbed, such as with antibiotic treatments, *C. difficile* is able to colonize and cause disease. *C. difficile* infection (CDI) especially problematic due to its burden on the healthcare system. Not only can *C. difficile* cause severe disease, such as toxic megacolon and death, but it is the primary cause of nosocomial diarrhea. CDI is primarily treated with antibiotics. Infections recalcitrant to antibiotics are eliminated by restoring the community with a fecal microbiota transplant (FMT), returning the perturbed community to a healthier protective state. However, FMTs are not always effective and have caused fatalities. Therefore we need to better treat and protect thus large population of susceptible people, antibiotic-treated patients, in an environment in with a high risk of *C. difficile* exsposure. Ideally, we need to identify susceptible patients and restore the colonization resistance of their gut microbiome.

In order to have the ability to identify susceptible microbial intestinal communities, we need to understand how the microbiome resists colonization. Previous research has identified potential mechanisms of colonization resistance such as bile salt metabolism and nutrient competition (Buffie, Theriot, Britton, Jenior). However, these studies have only been able to suppress the level of *C. difficile* colonization, not restore resistance. This disparity could be attributed to the complexity of the community and the mechanisms of colonization resistance. Previously, we have shown when *C. difficile* colonizes different antibiotic-treated murine communities, it modifies its metabolism to fit each specific environment @Schubert2015; @Jenior201, Jenior). Therefore, instead of using individudal mechanisms or microbes, we wanted to investigate the bacterial community dynamics when *C. difficle* is eliminated from differently susceptible communities. 

The purpose of this study was to improve our understanding of how gut bacterial communities prevent colonization through understanding the changes concurrent with elimination of *C. difficile* colonization. We hypothesize each colonized community has unique susceptibilities and requires specific changes to prevent and eliminate colonization. Jenior et al made the observation that Clindamycin-treated communities spontaneously clear *C. difficile* colonization whereas mice treated with Streptomycin and Cefoperazone do not. Then we modified the level of the later two antibiotics to permit *C. difficile* to colonize at a high level ~10^7 and then be spontaneously cleared from the community. With these three different communities, all with similar initial communities, able to spontaneously clear the colonization, we asked if the changes associated colonization and clearance were unique to the treatment. Using 16S rRNA gene sequencing to observe the community dynamics, we show each community had unique bacteria and community changes that were associated with clearance of *C. difficile* colonization. These data suggest that each case of CDI may need to be treated uniquely for the specific niche *C. difficile* has taken over.



## Results

**Clindamycin-treated communities spontaneously cleared *C. difficile* colonization**
Mice were treated with intraperitoneal injection of clindamycin (10 mg/kg) one day prior to challenge with *C. difficile*. Then, each day for 10 days, stool was collected for enumerating *C. difficile* CFUs and sequencing the 16S rRNA genes of the community. All mice were colonized to a high level and cleared the colonization within 10 days (Figure 1A). We then looked at the 16S rRNA sequences to understand the community changes occuring while *C. difficile* colonization was being reduced. The diversity of the community decreases greatly with antibiotic treatment and does not recover within the time period that *C. difficile* colonization is reduced (Figure 1B). However, the changes of the *C. difficile* challenged communities are similar to the changes observed in a mock challenged communities. Generally, we observe at the time of *C. difficile* challenge the community is dominated by *Escherichia*/*Shigella*, which decreases over the following six days, concurrent with reduction in *C. difficile* CFUs (Figure 1C). We see an opposite trend in *Porphyromonadaceae* and *Bacteroides* which are greatly reduced with antibiotics but recover as *C. difficile* colonization is reduced.

***C. difficile* is spontaneously cleared when reducing cefoperazone and streptomycin concentrations**
Mice treated with streptomycin or cefoperazone at concentrations of *C. difficile* infection models and challenged with *C. difficile*. These mice were  followed for 10 days, sampling stool daily for *C. difficile* CFU and bacterial 16S rRNA gene sequences to observe long-term colonization dynamics. In both experiments, *C. difficile* colonization was maintained (Figure 2A). Next, we repeated the *C. difficile* challenge with a two-step reduction in antibiotic concentration. In both antibiotic treatments, we observed the lowest concentration were either lowly and transiently colonized or not colonized (Figure 2B). Whereas, the intermediate concentration of both antibiotics resulted in a high level colonization in most mice and resulted in a mix of communities that cleared the colonization and others maintained the colonization. We next compared the 16S rRNA gene sequences of the communities at the time of infection. Cefoperazone treatment shifted the dominant community members from a community dominated by Porphyromonadaceae, Bacteroides and Akkermansia to one more dominated by Lactobacillus and Escherichia with the increasing antibiotic concentrations (Figure 2C). Streptomycin treatment maintained the dominant community genera Porphyromonadaceae, Bacteroides and Akkermansia but reduced most of the other genera of the community with increasing streptomycin concentration.

**Bacterial abundance changes associated with *C. difficile* colonization are unique to each treatment**
With these three different treatments of the same naive gut community, we investigated if the changes associated with clearance of *C. difficile* colonization we conserved. To begin, we compared the $\beta$-diversity $\theta$~YC~ dissimilarities of the cummunity structure at major time points (Figure 3A). The clindamycin-treated communities had a large shift with antibiotic treatment and slightly recovered by the time *C. difficile* was no longer colonized. Cefoperazone-treated communities mostly experienced a large shift in diversity with antibiotic treatment but were divergent in the recovery. Communities no longer colonized with *C. difficile* had shifted back to be most similar to their initial community structure, whereas the communities with *C. difficile* still present had little change in community structure from post-antibiotic treatment. Streptomycin-treatment communities exhibited no community structural differences between communities that remained colonized compared to those that had eliminated the colonization. These communities experienced a slight shift away from their initial community with antibiotic treatment and at the end of the experiment all returned to a community structure closely related to their initial community. We also compared the end point communities within and across treatments. Communites with other antibitoic treatments were generally more dissimilar in community structures than communities were to others with the same antibiotic treatment, with the exception of cefoperazone-treated communities that remained colonized. Cefoperazone-treated communities persistently colonized with *C. difficile* were greatly dissimilar with little differences at the end of the experiment. Overall, these data suggest returning to the initial community is critical for cefoperazone to clear colonization, but is not necessary for the other antibiotic treated communities.

Since these antibiotic-treated communities are different from each other we investigated what changes occur in each condition relative to *C. difficile* colonization. First, we compared the difference in abundance of bacterial operational taxonomic units (OTUs) between communities that ended with or without *C. difficile* present (Figure 3B). At the time of infection, cefoperazone-treated communities that remained colonized were significantly enriched in facultative anaerobes, *Streptococcus*, *Lactobacillus*, *Enterococcus*, *Pseudomonas*, *Staphylococcus* and *Escherichia*/*Shigella*. Whereas, cefoperazone-treated communities that had *C. difficile* colonization cleared had significant enrichment in eight different OTUs belonging to *Porphyromonadaceae* at the time of clearance. Streptomycin-treated communities had fewer differences between outcomes. Communities that remained colonized were overall depleted of *Alistipes* and at time of infection were depleted of two *Porphyromonadaceae* OTUs and *Anaeroplasma*. Next, we determined the significant changes occurring between major time points within each colonization outcome (Figure 3C). Clindamycin-treated communities showed large decreases in 23 OTUs and a near complete dominant bloom of *Escherichia/Shigella*. With the elimination of *C. difficile*, we observed a drastic reduction of *Escherichia*/*Shigella* and recovery of 9 OTUs belonging to *Porphyromonadaceae*, *Bacteroides*, *Akkermansia*, *Lactobacillus*, *Bifidobacterium* and *Lachnospiraceae*. The differences in cefoperazone-treated communities were primarily observed in persistently colonized communities. These communities had a loss of numerous *Porphyromonadaceae* OTUs and increases in the facultative anaerobes differentiating the cefoperazone-treated communities by outcome (Figure 3B). Lastly, the temporal changes in streptomycin-treated communities were more subtle. Most of the differences were observed with antibiotic treatment and many of those differences were recovered by the end of the experiment. Generally, the communities that were no longer colonized had more increased abundance of *Porphyromonadaceae* OTUs and fewer depleted OTUs at the time of infections. At the end of the experiment various OTUs shifted towards 1% relative abundance in communities without *C. difficile*, whereas in when *C. difficile* persisted *Olsenella* was depleted and three *Porphyromonadaceae* OTUs increased to nearly 10% relative abundance. Together, these data reveal antibiotic-treatment specific dynamics associated with the elimination of *C. difficile* colonization.

**Distinct features of the bacterial community at the time of infection predict end point colonization**
With the observation distinct community differences occur with clearance for each treatment, we sought to determine if the community composition at the time of *C. difficile* challenge can correctly classify the colonization outcome. We employed a machine learning model, L~2~ Logistic Regression, to determine the OTUs contributing most to differentiating the colonization outcome (Figure 4). Our model was able to correctly classify our samples with an AUROC of **0.9XX** (Figure S1). While this model is likely overfit due to the strong correlation between cage effects and outcome, we are interested in the features driving the classification. We next randomly permuted each OTU feature to determine its contribution to correctly classifying the outcome. This analysis revealed OTUs from our previous analyses in addition to the information of other community members and the relationship between their abundances is able to correctly classify the outcome. Thus, high abundance of an OTU such as *Escherichia*/*Shigella* does not necessarily indicate outcome, but including the abundance of other community members allows for accurate outcome classification. Therefore, the model is relying on contributions from multiple OTUs to classify the experimental outcome. These data can offer a basis to hypothesize the distinct combinations of bacteria promoting the ability to eliminate *C. difficile* colonization.
    
**Conditional independence networks reveal treatment-specific relationships between the community and *C. difficile* during colonization clearance**
*Escherichia*/*Shigella* abundance appears to have community dependent effects on determining outcome, however we know that the changes in abundance after the time of *C. difficile challenge differ. So next, we inquired how the changes in *C. difficile* colonization related to the changes in bacterial abundance to elucidate the bacteria directly affecting the colonization level of *C. difficile*. We developed a conditional independence network for each treatment using SPIEC-EASI (sparse inverse covariance estimation for ecological association inference) to uncover which OTUs are specifically interactin g with *C. difficile* and not merely correlated.  First, we focused on the first-order connections to *C. difficile* (Figure 5A). We observed *C. difficile* has limited interactions in cefoperazone-treated communities, the primary relationship being a positive with *Escherichia*/*Shigella*. In clindamycin-treated communities, *C. difficile* colonization has positive relationships with *Escherichia*/*Shigella*, *Pseudomonas*, and *Olsenella* and negative relationships with *Lachnospiraceae* and *Clostridium* XIVa. In streptomycin-treated communities, *C. difficile* has negative relationships with *Porphyromonadaceae* and positive relationships with *Ruminococcaceae*, *Bacteroidetes*, *Clostridium* IV and *Olsenella*. Finally, we also compared the whole network (Figure 5B). This analysis revealed cefoperazone treatment results in a network with much lower degree centrality, composed of primarily single connections. Additionally, cefoperazone has significantly lower betweenness centrality and those communities which *C. difficile* persists have even lower betweenness centrality. Therefore, unique subsets of bacteria interact with *C. difficile* in each antibiotic-treated communities and  cefoperazone treatment eliminates many bacteria critical to community wide interactions.



## Discussion 

Previous studies investigating how gut microbial communities resist *C. difficile* colonization have attempted to restore colonization resistance through addressing mechanisms identified through collective analyses of susceptible communities. Here, we elucidated community-specific changes concurrent with the clearance of *C. difficile* colonization. Even though these communities all began with closely related communities, colonization clearance occurred with treatment-specific changes and returning to their baseline community was not necessary. These data suggest that the same treatment to restore colonization resistance may not be effective in all cases. A specific treatment designed for the specific susceptibilities will be most effective to prevent and eliminate *C. difficile* colonization. Further investigation into specific cases of *C. difficile* colonization will elucidate the niches *C. difficile* utilizes and the interventions necessary to prevent or eliminate *C. difficile* colonization.

In this study we have shown antibiotic treatment resulted in specific changes for both *C. difficile* colonization and its clearance. Clindamycin-treated communities were susceptible with a dominant bloom in *Escherichia*/*Shigella* which naturally resolved. In this scenario clearance was associated with the recovery some of the bacteria populations reduced with antibiotic treatment. However, the community recovery was similar between *C. difficile*-challenged and mock-challenged communities. Therefore, the clindamycin-treated communities may naturally restore colonization resistance within a short time period. If there is a greater delay between clindamycin treatment and *C. difficile* challenge, *C. difficile* may be prevented from colonizing. Streptomycin-treated communities were susceptible with minor changes in diversity and only the change in abundance of relatively few bacteria and subtle differences between colonization outcomes. It is plausible in this scenario susceptiblity is driven by the loss of a few critical bacteria. Many of the important bacteria associated with differences in colonization were *Porphyromonadaceae* and communities that clear colonization maintained more OTUs of *Porphyromonadaceae*. However, the abunance of this genus did not differ greatly. This conflict led us to the hypothesis that multiple species of *Porphyromonadaceae* are necessary to provide colonization resistance. Colonization resistance could potentially be restored through returning the diversity of *Porphyromonadaceae*. Cefoperazone-treated communities became susceptible with the expansion of numerous facultative anaerobes while suppressing many of the initially dominant genera. Communites able to clear were able to recover some of their original community. In this scenario the community appears to be inhibited from clearing the colonization due to the sustained presence of the facultative anaerobes. While all of these communities had the similar initial communities, each of these communities were susceptible to and cleared *C. difficile* colonization in different ways.

While we have described treatment-specific changes, the bacterial changes observed here are not likely to directly transfer to human studies. Murine gut communities differ by their care facility and even greater from human gut communities @source. It is not likely bacteria identified here could translate to human interventions. Therefore, we approached the data more wholistically than focusing on the specific OTUs in the hopes to understand the general trends as opposed to the experiment-specific events. We categorized the general changes, microbial relationships and potential interventions of these colonization scenarios in to three themes. (1) Right place, right time - a transient dominance by a facultative anaerobe permits *C. difficile* colonization but quickly recovers and *C. difficile* is not able to persist. Since this susceptibility is due to a transient repression of the community, interventions could possibly make the infection worse. If time is sufficient for the community to recover and eliminate infection, treating the community with an antibiotic or the preparation for an FMT may create new susceptibilites and eliminate other protective fuctions or open new nutrient niches. This unnecessary intevention could lead to increased susceptiblility, allowing *C. difficile* to spread to other niches and prolong the infection. This susceptibility may be identified by a disappearing facultative anaerobe bloom. (2) Single meal option menu, take it or leave it - a perturbation only affects a select portion of the microbiome, leading to minor abundance changes and a slight decrease in diversity, opening a limited specific niche for *C. difficile* to colonize. A few specific bacteria would be necessary to recolonize the exposed niche and eliminate *C. difficile* colonization. A fecal microbiota transplant may seed the necessary bacteria to clear *C. difficile* from the exposed niche of the colonized community. (3) All day all-you-can-eat buffet - a significant perturbation occurs leading to a persistent increase in facultative anaerobes and many different niches become available. *C. difficile* is able to colonize numerous niches. A full FMT may be sufficient but this scenario is sensitive to there being sufficient diversity of the transplanted community to recover all the susceptibilites. These communities have been severely depleted and are likely lacking numerous protective functions and lack any bacteria that overlap in abilities. It may be necessary to do multiple FMTs or transplant a functionally-enriched fecal community to recover the community sufficiently enough to outcompete *C. difficile* for nutrient niches and replace missing protective functions. Analyzing each of these colonization scenarios individually reveals how uniquely each susceptible and recovery happens.

Based on these observations, future investigations can further elucidate the uniqueness of susceptible communities and the requirements to restore colonization resistance. One common theme for susceptibility across treatments was increased facultative anaerobe abundance. This increase could be investigated for its ability to predict community susceptiblilty to *C. difficile* colonization. Facultative anaerobe blooms is liekly related to the loss of oxygen level suppression by obligate anaerobes. However, it is unclear in certain cases what prevents the succession of the primary community members from facultative anerobes back to obligate anaerobes. Future studies should further investigate this relationship between facultative anaerobe blooms and susceptibility and interventions to assist the community to recover the obligate anaerobes. Here we only used one strain of *C. difficile* in our experiments. We expect our general themes to hold true for all strains, although it is possible other strains will have different community interactions. For example, more virulent strains, like *C. difficile* VPI 10463, may affect the gut environment more since it produces more toxin. This change could impact the susceptible community and the bacterial community may have requirements to clear *C. difficile*. While here we have shown the communities changes to clear colonization are unique and we have previously shown community function at peak infection are different, it is still unclear if the community functions reducing C. difficile populations are the same. The changes in transcription and metabolites will greater illuminate the necessary changes to eliminate *C. difficile*. Building upon the information presented in this study will create a clearer picture of how *C. difficile* colonizes out intestinal community.

Here we have shown murine gut communities can become uniquely susceptible to *C. difficile* colonization and then subsequently uniquely clear the colonization. This set of experiments shows that one treatment for all *C. difficile* infections may not be appropriate. While our current use of FMT to eliminate recurrent CDI is highly effective, it does not work in all patients and has even resulted in adverse consequences @source. The findings in this study may help explain why FMTs may be ineffective. While the FMT is transplanting a whole community, some members and functions will be lost through the aerobic preparation and transfer process, potentially bacteria necessary to fill a subset of patients susceptibilities. Furthermore this study demonstrates a method to improve our understanding of how *C. difficile* colonizes communities. With the knowledge of how a community is susceptible and how *C. difficile* responds to the gut microbiome will lead to targeted therapies to better management of CDI.



## Materials and Methods

**Animal care**
All mice were obtained from a single breeding colony and maintained in specific-pathogen-free (SPF) conditions at the University of Michigan animal facility. All mouse protocols and experiments were approved by the University Committee on Use and Care of Animals at the University of Michigan and completed in agreement with approved guidelines.

**Antibiotic administration**
Mice were given one of three antibiotics, cefoperazone, clindamycin, or streptomycin. Cefoperazone (0.5, 0.3, or 0.1 mg/ml) and streptomycin (5, 0.5, or 0.1 mg/ml) were delivered via drinking water for 5 days. Clindamycin (10 mg/kg) was administered through intraperitoneal injection. 

***C. difficile* challenge**
Mice were returned to untreated drinking water for 24h before challenging with *C. difficile* strain 630Î”erm spores. *C. difficile* spores were aliquoted from a single spore stock stored at 4â„ƒ. Spore concentration was determined one week prior to the day of challenge (). 103 *C. difficile* spores were orally gavaged into each mouse. Once the gavages were completed, the remaining spore solution was serial diluted and plated to confirm accurate spore concentration was delivered.

**Sample collection**
Fecal samples were collected on the day antibiotic treatment was started, on the day of *C. difficile* challenge and the following 10 days. For the day of challenge and beyond, a fecal sample was also collected and  weighed. Then under anaerobic conditions a fecal sample was serial diluted in anaerobic phosphate-buffered saline and plated on TCCFA plates. After 24-h of anaerobic incubation at 37â„ƒ CFU were enumerated ().

**DNA sequencing**
Total bacterial DNA was extracted from each fecal sample using MOBIO PowerSoil-htp 96-well soil DNA isolation kit. We created amplicons of the 16s rRNA gene V4 region and sequenced them using an Illumina MiSeq as described previously ().

**Sequence curation**
Sequences were processed using mothur(v.1.XX) as previously described (). Briefly, we used a 3% dissimilarity cutoff to group sequences into operational taxonomic units (OTUs). We used a naive Bayesian classifier with the Ribosomal Database Project training set (version XX) to assign taxonomic classifications to each OTU (). With the fecal samples, we also sequenced a mock community with a known community composition and their true 16s rRNA gene sequences. We processed this mock community along with our samples to determine our sequence curation resulted in an error rate of X.XX%

**Statistical analysis and modeling**
Diversity comparisons were calculated in mothur. To compare within community diversty, we calculated Sobs and Inverse Simpson. To compare across communities, we calculated dissimilarity matrices based on metric of Yue and Clayton (). Samples were rarified to 2,000 sequences per sample within calculating the distance matrices to ensure no biases remained due to uneven sampling.  OTUs were subsampled to 2,000 counts per sample and remaining statistical analysis and data visualization was performed in R (v#.#.#) with the tidyverse package (v#.#.#). Pairwise OTU-to-OTU comparisons were calculated by pairwise Wilcoxon rank sum test and then P values were corrected for multiple comparisons with a Benjamini and Hochberg adjustment for a type I error rate of 0.05 (). Logistic regression models were contructed using half the data to train and the other to test the model.  The model was developed from the caret R package and previously developed machine learning pipeline @Topcuoglu2020; @. Conditional independence networks were calculated using  SPIEC-EASI (sparse inverse covariance estimation for ecological association inference) methods from the SpiecEasi R package after optimizing lambda to 0.001 with a network stability between 0.045 and 0.05 (). Network centrality measures degree and betweenness were calculated on whole networks using functions from the igraph R package ().

**Code availability**
Scripts necessary to reproduce our analysis and this paper are available in an online repository (https://github.com/SchlossLab/Lesniak_Clearance_XXXX_2020).

**Sequence data accession number**
All 16s rRNA gene sequence data and associated metadata are available through the Sequence Read Archive via accession no. SXX#######.

## Acknowledgements 

Thank you to BegÃ¼m TopÃ§uoglu and Sarah Tomkovich for critical discussion in the development and execution of this project.
This work was supported by several grants from the National Institutes for Health **R01GM099514, R01HG005975, U19AI090871, and P30DK034933. Additionally, NAL was supported by the Molecular Mechanisms of Microbial Pathogenesis training grant (NIH T32 AI007528).**
The funding agencies had no role in study design, data collection and analysis, decision to publish, or preparation of the manuscript.


\newpage

![Figure 1](../results/figures/figure_1.jpg)

**Figure 1. Clindamycin-treated murine communities spontaneously clears *C. difficile* colonization** (A) Daily CFU of *C. difficile* in fecal samples of clindamycin-treated mice from time of challenge (Day 0) through 10 days post infection (dpi). The bold line is the median CFU of the group and the transparent lines are the individual mice.  (B) Quantification of diversity over time from pre-treatment (Day -1) through 10 dpi. Red colored boxes were challenged with *C. difficile* strain 630, whereas gray boxes were mock-challenged. (C) Daily relative abundance of ten most abundant genera, all other genera grouped into Other. Color intensity is log~10~-transformed mean percent relative abundance of each day. (N = 11 mice). Limit of detection (LOD). Boxplot shows median - horizontal bar, quartile range - box, range of non-outlier data - whiskers, data greater than 1.5 times the interquartile range - points.


![Figure 2](../results/figures/figure_2.jpg)

**Figure 2. *C. difficile* is spontaneously cleared when reducing cefoperazone and streptomycin concentrations** (A) Daily CFU of *C. difficile* in fecal samples of cefoperazone and streptomycin-treated mice from time of challenge (Day 0) through 10 dpi. Antibiotic concentrations used were *C. difficile* infection model levels (0.5 mg/mL and 5.0 mg/mL). (B) Daily CFU of *C. difficile* in fecal samples of cefoperazone and streptomycin-treated mice from time of challenge (Day 0) through 10 days post infection (dpi). Antibiotic concentrations used were two-step decreases from *C. difficile* infection model levels (0.3 and 0.1 mg/mL and 0.5 and 0.1 mg/mL, respectively). (C) Relative abundance of twelve most abundant genera on the day of *C. difficile* challenge (Day 0), all other genera grouped into Other. Color intensity is log10-transformed mean percent relative abundance of each day. (cefoperazone - 0.5 mg/mL N = 5, 0.3 mg/mL N = 9, 0.1 mg/mL N = 2; streptomycin - 5.0 mg/mL N = 8, 0.5 mg/mL N = 7, 0.1 mg/mL N = 7). Limit of detection (LOD). Bold line is the median CFU of the group and transparent lines are the individual mice.


![Figure 3](../results/figures/figure_3.jpg)

**Figure 3. Individual bacteria abundance changes associated with *C. difficile* colonization are unique to each treatment** (A) $\beta$-diversity $\theta$~YC~ comparisons - Initial vs Initial compares one pre-antibiotic treatment to all other pre-antibiotic treatment samples; Initial vs Time of Infection (TOI) and End vs Initial compare time points within mouse, intra-group comparisons compare a sample to all other samples receiving the same antibiotic treatment and inter-group comparisons compares a sample to all other samples not receiving the same antibiotic treatment mouse to all other. Boxes colored by time point being compared (initial = green; time of infection (TOI) = blue; end = red) to the same time point or to the initial time point. Comparisons restricted to within antibiotic and outcome except inter-group comparisons. (B) The difference in the relative abundance of OTUs that are significantly different between communities that were no longer colonized by *C. difficile* and those that remained samples colonized within each antibiotic treatment for each timepoint. Red points - community cleared colonization. Blue point - community remianed colonized. (C) The difference in the relative abundance of OTUs that are significantly different communities between time points within each *C. difficile* colonization outcome for each antibiotic treatment. Initial points - green; TOI points - blue; End points - red. Arrows point in the direction of the temporal change of the relative abundance. Lines connect points by OTU within each comparison to show difference in medians. Bold OTUs were shared across comparisons. All OTUs plotted in panel B and C were significantly different, calculated by Wilcoxon rank sum test with Benjamini-Hochberg correction. Bold points are median relative abundance and transparent points are relative abundance of individual mice. Limit of detection (LOD). Boxplot shows median - horizontal bar, quartile range - box, range of non-outlier data - whiskers, data greater than 1.5 times the interquartile range - points.


![Figure 4](../results/figures/figure_4.jpg)

**Figure 4. Distinct features of the bacterial community at the time of infection can classify end point colonization** (A) L~2~ logistic regression model feature importances determined by the negative effect randomizing an individual feature has on the model performance. All OTUs affecting performance shown. (C) Relative abundance difference in features used by L~2~ logistic regression model faceted by antibiotic treatment. Red boxplot - antibiotic-treated communities that were no longer colonized; Blue boxplot - antibiotic-treated communities that remained colonized. Boxplot shows median - horizontal bar, quartile range - box, range of non-outlier data - whiskers, data greater than 1.5 times the interquartile range - points.



![Figure 5](../results/figures/figure_5.jpg)
**Figure 5. Conditional independence networks reveal treatment-specific relationships between the community and *C. difficile* during colonization clearance** (A) SPIEC-EASI (sparse inverse covariance estimation for ecological association inference) networks showing conditionally independent first-order relationships between *C. difficile* and the community as *C. difficile* is cleared for the gut environment. Nodes sized by median relative abundance of OTU. Edges thickness is determined by the relationship strength and colored red for negative interactions and blue for positive interactions. (B) Network centrality measured with betweenness, how many paths between two OTUs pass through an individual, and degree, how many connections an OTU has. Colored by antibiotic (cefoperazone - blue; clindamycin - red; streptomycin - yellow) and boxplots are filled in gray for networks that remained colonized. Boxplot shows median - horizontal bar, quartile range - box, range of non-outlier data - whiskers, data greater than 1.5 times the interquartile range - points.


![Figure S1](../results/figures/figure_S1.jpg)
**Figure S1. Bacterial community at the time of infection can classify endpoint colonization** Classification performance of L~2~ logistic regression. Area under the receiver-operator curve for classifying if the community will remain colonized based on the OTUs present at the time of *C. difficile* infection (Day 0). 


\newpage

## References